Description: Generate stronger server SSL/TLS certificate to be OpenSSL 1.1.0 compatible
Forwarded: https://jira.mariadb.org/browse/MDEV-18019
Author: Otto Kekäläinen <otto@mariadb.org>
Last-Update: 2018-12-16

 Since OpenSSL 1.1.0 the mysqld will refuse connections due to this OpenSSL error:
 [Warning] SSL error: Unable to get certificate
 [Warning] SSL error: error:140AB18F:SSL routines:SSL_CTX_use_certificate:ee key too small

 By generating a new stronger certificate the server will be operational again.


--- a/mysql-test/lib/generate-ssl-certs.sh
+++ b/mysql-test/lib/generate-ssl-certs.sh
@@ -9,13 +9,16 @@ cd std_data/
 rm -rf demoCA
 mkdir demoCA demoCA/newcerts
 touch demoCA/index.txt
+touch demoCA/index.txt.attr
 echo 01 > demoCA/serial
 
+# OpenSSL 1.1.0 requires by default stronger keys, so use rsa:4096 at minimum for all keys
+
 # CA certificate, self-signed
-openssl req -x509 -newkey rsa:2048 -keyout cakey.pem -out cacert.pem -days 7300 -nodes -subj '/CN=cacert/C=FI/ST=Helsinki/L=Helsinki/O=MariaDB' -text
+openssl req -x509 -newkey rsa:4096 -keyout cakey.pem -out cacert.pem -days 7300 -nodes -subj '/CN=cacert/C=FI/ST=Helsinki/L=Helsinki/O=MariaDB' -text
 
 # server certificate signing request and private key. Note the very long subject (for MDEV-7859)
-openssl req -newkey rsa:1024 -keyout server-key.pem -out demoCA/server-req.pem -days 7300 -nodes -subj '/CN=localhost/C=FI/ST=state or province within country, in other certificates in this file it is the same as L/L=location, usually an address but often ambiguously used/OU=organizational unit name, a division name within an organization/O=organization name, typically a company name'
+openssl req -newkey rsa:4096 -keyout server-key.pem -out demoCA/server-req.pem -days 7300 -nodes -subj '/CN=localhost/C=FI/ST=state or province within country, in other certificates in this file it is the same as L/L=location, usually an address but often ambiguously used/OU=organizational unit name, a division name within an organization/O=organization name, typically a company name'
 # convert the key to yassl compatible format
 openssl rsa -in server-key.pem -out server-key.pem
 # sign the server certificate with CA certificate
@@ -25,7 +28,7 @@ openssl req -newkey rsa:8192 -keyout ser
 openssl rsa -in server8k-key.pem -out server8k-key.pem
 openssl ca -keyfile cakey.pem -days 7300 -batch -cert cacert.pem -policy policy_anything -out server8k-cert.pem -infiles demoCA/server8k-req.pem
 
-openssl req -newkey rsa:1024 -keyout client-key.pem -out demoCA/client-req.pem -days 7300 -nodes -subj '/CN=client/C=FI/ST=Helsinki/L=Helsinki/O=MariaDB'
+openssl req -newkey rsa:4096 -keyout client-key.pem -out demoCA/client-req.pem -days 7300 -nodes -subj '/CN=client/C=FI/ST=Helsinki/L=Helsinki/O=MariaDB'
 openssl rsa -in client-key.pem -out client-key.pem
 openssl ca -keyfile cakey.pem -days 7300 -batch -cert cacert.pem -policy policy_anything -out client-cert.pem -infiles demoCA/client-req.pem
 
@@ -33,7 +36,7 @@ openssl ca -keyfile cakey.pem -days 7300
 cat > demoCA/sanext.conf <<EOF
 subjectAltName=DNS:localhost
 EOF
-openssl req -newkey rsa:1024 -keyout serversan-key.pem -out demoCA/serversan-req.pem -days 7300 -nodes -subj '/CN=server/C=FI/ST=Helsinki/L=Helsinki/O=MariaDB'
+openssl req -newkey rsa:4096 -keyout serversan-key.pem -out demoCA/serversan-req.pem -days 7300 -nodes -subj '/CN=server/C=FI/ST=Helsinki/L=Helsinki/O=MariaDB'
 openssl ca -keyfile cakey.pem -extfile demoCA/sanext.conf -days 7300 -batch -cert cacert.pem -policy policy_anything -out serversan-cert.pem -infiles demoCA/serversan-req.pem
 
 rm -rf demoCA
--- a/mysql-test/std_data/cacert.pem
+++ b/mysql-test/std_data/cacert.pem
@@ -2,78 +2,120 @@ Certificate:
     Data:
         Version: 3 (0x2)
         Serial Number:
-            e5:b1:e3:71:e9:6f:a9:e1
+            ff:0e:52:2f:3f:e2:c0:ff
     Signature Algorithm: sha256WithRSAEncryption
-        Issuer: CN=cacert, C=FI, ST=Helsinki, L=Helsinki, O=MariaDB
+        Issuer: CN = cacert, C = FI, ST = Helsinki, L = Helsinki, O = MariaDB
         Validity
-            Not Before: Apr 25 20:52:21 2017 GMT
-            Not After : Apr 20 20:52:21 2037 GMT
-        Subject: CN=cacert, C=FI, ST=Helsinki, L=Helsinki, O=MariaDB
+            Not Before: Dec 16 17:57:00 2018 GMT
+            Not After : Dec 11 17:57:00 2038 GMT
+        Subject: CN = cacert, C = FI, ST = Helsinki, L = Helsinki, O = MariaDB
         Subject Public Key Info:
             Public Key Algorithm: rsaEncryption
-                Public-Key: (2048 bit)
+                Public-Key: (4096 bit)
                 Modulus:
-                    00:a0:ad:d5:b1:ec:45:6f:d6:33:fc:5a:03:29:14:
-                    f1:8e:78:d5:27:53:79:e0:92:7c:10:3b:79:a0:d7:
-                    b6:9d:a8:5c:4d:fa:68:11:b3:03:9e:ee:5e:20:79:
-                    23:d8:9c:49:34:9c:1d:c4:6e:53:1f:9a:92:1f:08:
-                    c1:15:e2:ad:cf:59:cd:1e:55:84:79:f9:09:ca:36:
-                    8a:50:83:c6:38:48:c6:d3:fa:f6:f2:2a:4f:bd:5d:
-                    60:9d:eb:21:c4:8c:f2:dd:2d:49:10:63:46:47:de:
-                    2d:59:a0:4a:e0:58:e6:c0:ae:d8:d4:5e:9a:f8:f5:
-                    68:1d:ea:80:8a:d6:01:b0:d5:5f:30:4d:88:5a:c5:
-                    1f:81:92:c1:40:54:c8:bb:a6:a1:43:de:81:3c:4b:
-                    79:95:82:bb:52:da:a3:a4:a0:69:ff:7e:00:8c:86:
-                    85:ec:af:03:68:a8:83:48:a0:e4:1d:31:a9:5c:47:
-                    99:9d:3a:3f:b5:3e:12:7c:4d:47:15:72:f1:11:5c:
-                    4a:ef:08:1c:7b:8f:e6:03:06:07:4f:94:21:b0:5e:
-                    27:fa:93:8c:b4:cc:56:34:3b:6d:c4:4a:14:57:b2:
-                    21:1a:3e:2f:c5:9e:47:1a:59:05:22:0e:56:b1:a7:
-                    e8:80:9b:82:c3:54:57:12:05:94:79:a2:03:d9:64:
-                    3c:63
+                    00:c3:9f:e8:e5:e2:ba:35:58:6f:5d:a2:97:71:e1:
+                    04:29:27:c8:67:d8:12:7f:cb:86:e4:de:c7:06:21:
+                    34:7d:fa:1f:cb:41:f7:21:b9:a9:e8:87:97:d1:a6:
+                    92:d0:3f:20:9f:81:b2:62:63:59:c1:d5:c5:1f:9b:
+                    86:0e:3c:67:62:dd:01:fd:ea:d3:db:55:10:06:5a:
+                    52:38:ea:b9:e7:fa:af:74:90:75:65:c6:87:df:42:
+                    8b:c3:0b:02:0f:96:9d:58:3b:71:df:00:09:56:9d:
+                    d3:52:2b:80:f2:3f:b1:cd:3e:69:b9:e1:28:cb:c8:
+                    58:d9:d2:bb:b8:fd:f9:9a:cf:46:28:b4:88:5c:90:
+                    34:c2:88:82:18:ec:e0:46:de:d1:f1:99:35:b6:bd:
+                    9f:ad:d0:87:3d:a3:6c:e3:b0:01:f0:49:ca:71:4d:
+                    84:07:ab:97:4b:bf:b0:b3:17:6f:29:30:d9:07:d1:
+                    a4:49:ea:da:7b:c1:50:5e:00:54:f8:cd:79:e4:d4:
+                    21:31:d0:1b:b2:3c:38:30:0b:66:bd:f9:07:bc:fd:
+                    76:ea:cf:ee:84:7a:52:98:5b:9c:2f:11:11:a6:3f:
+                    63:53:39:ed:c3:e8:84:56:c7:5f:47:6e:bc:5b:1b:
+                    1f:5f:70:c0:90:49:5e:91:b9:20:90:7e:5f:78:37:
+                    fc:84:67:97:31:fb:b2:8e:84:a5:24:3d:83:9f:bf:
+                    19:4b:64:02:3a:e8:ba:ae:fd:f2:0c:a6:a3:9f:f3:
+                    cc:e9:be:6e:30:27:0a:e8:9e:ad:38:d4:ab:10:03:
+                    b4:28:ba:c5:d1:c0:cb:51:8b:ed:ae:34:1e:74:1d:
+                    e0:1b:77:a9:da:02:67:8a:58:97:47:ef:31:66:25:
+                    cf:ef:e1:4d:72:50:92:c3:65:cc:59:49:1a:c7:b1:
+                    b9:e6:91:70:05:80:ce:61:9d:27:a0:88:0d:a4:e4:
+                    89:c3:e9:e9:68:b6:6c:1a:b0:a2:9a:39:06:fb:b9:
+                    57:4f:1f:91:8c:bd:60:61:30:f8:88:d3:ea:7d:56:
+                    a2:85:09:71:70:50:d5:6e:89:2d:b8:d9:e5:df:4c:
+                    0d:ca:b8:49:dc:5d:21:4c:d7:4f:f6:22:b3:80:1f:
+                    da:2d:8f:d7:9c:cf:4a:66:1e:01:96:95:92:df:4e:
+                    b0:c8:47:f3:4b:0d:40:af:af:c2:66:f4:41:63:a5:
+                    9e:f2:07:a9:7d:6f:3a:be:0c:a3:15:5c:e5:f0:72:
+                    7c:af:28:a7:fe:36:c9:eb:1f:f4:73:79:0e:44:9f:
+                    4b:8e:66:07:a3:62:8a:51:eb:f6:66:64:49:d9:42:
+                    a1:19:db:f8:79:65:5b:b4:e0:e1:80:48:66:09:c9:
+                    fe:f9:05
                 Exponent: 65537 (0x10001)
         X509v3 extensions:
             X509v3 Subject Key Identifier: 
-                1C:C7:2B:AA:1B:B1:BB:2E:9A:F4:0F:B1:86:60:57:38:C2:41:05:12
+                DA:2C:A1:63:9E:33:1D:44:F2:E1:37:17:C9:8F:90:4F:69:F8:B0:85
             X509v3 Authority Key Identifier: 
-                keyid:1C:C7:2B:AA:1B:B1:BB:2E:9A:F4:0F:B1:86:60:57:38:C2:41:05:12
+                keyid:DA:2C:A1:63:9E:33:1D:44:F2:E1:37:17:C9:8F:90:4F:69:F8:B0:85
 
-            X509v3 Basic Constraints: 
+            X509v3 Basic Constraints: critical
                 CA:TRUE
     Signature Algorithm: sha256WithRSAEncryption
-         0d:4b:21:52:fa:49:34:56:14:db:83:ae:1c:3d:a7:4d:3e:ea:
-         55:7e:1a:37:7a:65:89:ee:19:05:94:9d:3a:ad:59:c4:38:16:
-         b2:bd:02:ee:5a:a6:7e:e2:b1:21:a3:ad:af:8c:ae:c3:30:71:
-         ad:d7:d2:24:0f:c4:d9:47:80:c5:95:05:1d:7c:8a:49:0a:7d:
-         8b:61:ca:b5:68:3d:3e:4e:f1:c7:45:62:c8:cc:a9:2f:f3:12:
-         f1:3f:92:34:7f:07:ab:d3:ac:ab:af:2d:c9:69:63:8a:b2:e5:
-         35:ea:7d:b8:17:38:72:82:5f:96:3d:dc:8d:e5:11:bb:ae:f3:
-         02:2d:20:77:5c:64:59:18:a6:e7:fa:c7:89:e8:30:12:14:04:
-         40:5b:e9:b1:8f:86:81:b9:0d:6c:b6:fc:98:f9:b7:52:ab:8f:
-         7e:53:c8:a0:05:e4:cd:0d:6b:d2:74:9f:17:7a:a1:c3:76:5e:
-         f3:29:1c:c6:be:56:ab:02:f7:5d:e1:c9:21:27:6d:66:7a:41:
-         29:49:a3:f8:f5:2a:e7:03:2a:7c:52:4b:f5:46:58:45:be:a4:
-         4c:a0:65:37:1d:d8:ac:f8:1f:81:ca:9c:79:f0:ff:22:8c:1d:
-         ce:2b:d0:1e:ce:99:f2:db:fa:66:84:e6:86:6f:19:3b:10:f1:
-         92:ac:57:b2
+         6a:4f:16:3d:70:75:52:88:75:85:c8:39:29:a6:b8:58:80:c6:
+         85:e0:cd:24:4c:15:31:5a:42:19:cf:b3:b6:4e:59:5e:cf:fd:
+         55:3f:38:09:c9:a8:f9:c1:10:d7:25:68:7d:7b:d9:0a:e5:74:
+         34:03:56:37:79:09:12:14:9b:0e:af:16:c4:99:3d:07:53:39:
+         8f:23:73:fa:8d:3e:af:f0:45:36:c6:a1:71:a9:0a:22:f7:f2:
+         86:b3:46:e4:a6:59:48:a8:36:fb:f6:5e:6c:0e:9f:0d:0e:c2:
+         32:c9:1b:a4:d3:07:6a:61:0d:37:28:96:b0:e8:5c:d5:39:16:
+         da:12:47:c7:86:05:31:83:83:d3:f0:ea:92:39:22:6a:2e:73:
+         da:46:b1:fa:c6:19:a3:1a:1e:de:67:e0:57:74:58:7a:0d:d7:
+         a5:7e:c4:e3:68:67:55:00:ae:f6:28:89:1e:c1:25:8f:69:24:
+         bc:5f:fc:f8:34:5a:de:9e:7b:99:3d:d8:8a:d9:3d:af:a4:0a:
+         71:00:66:52:8f:94:08:bf:69:09:48:ec:f1:e0:18:84:ee:f3:
+         9f:9f:e5:51:44:3b:e6:56:70:f5:5f:f9:25:b9:44:2a:dd:89:
+         28:90:c2:f2:0a:36:10:d4:07:79:11:48:7f:49:0d:60:09:71:
+         6a:6a:a5:72:6d:11:a0:33:48:76:0d:eb:e6:dc:01:c3:06:48:
+         fb:3a:d4:70:73:38:dd:6b:18:13:10:ca:ab:f6:c0:c1:f0:f1:
+         9b:61:3d:1b:7c:03:52:4d:24:8a:69:07:10:a7:21:b8:81:18:
+         e2:df:54:52:f7:9b:fa:10:ec:b3:35:08:84:84:42:48:42:e2:
+         ed:30:8e:aa:20:f2:23:ca:c1:ff:30:91:01:fe:07:a4:8e:c2:
+         30:09:fd:78:7e:68:c2:34:6d:d1:b1:12:5e:c1:18:1c:f4:a6:
+         db:68:96:d7:5c:ff:c7:6e:a7:e4:cf:7f:43:d0:fe:77:7d:51:
+         8b:48:41:25:55:86:dc:a1:9e:7a:32:63:27:1e:24:00:da:2f:
+         a0:28:78:55:65:2d:37:6c:c2:53:e1:be:d2:54:a6:a0:1b:20:
+         87:0c:bd:12:18:5e:76:63:71:81:b1:fc:f3:e0:f1:dc:3d:ad:
+         b6:c0:32:0f:e4:35:ca:14:0c:6b:5a:a2:4d:4a:6e:08:b0:66:
+         e4:24:3b:e8:72:47:d0:4f:23:29:61:b6:10:87:6c:42:22:5b:
+         bb:82:d6:3e:5f:e0:13:9c:d6:38:11:23:a4:bb:3f:f1:56:44:
+         74:32:b5:b8:d9:19:e3:71:9f:98:49:0d:a5:e3:b6:6c:a9:4f:
+         fa:f3:29:75:35:9a:04:1b
 -----BEGIN CERTIFICATE-----
-MIIDfzCCAmegAwIBAgIJAOWx43Hpb6nhMA0GCSqGSIb3DQEBCwUAMFYxDzANBgNV
+MIIFgjCCA2qgAwIBAgIJAP8OUi8/4sD/MA0GCSqGSIb3DQEBCwUAMFYxDzANBgNV
 BAMMBmNhY2VydDELMAkGA1UEBhMCRkkxETAPBgNVBAgMCEhlbHNpbmtpMREwDwYD
-VQQHDAhIZWxzaW5raTEQMA4GA1UECgwHTWFyaWFEQjAeFw0xNzA0MjUyMDUyMjFa
-Fw0zNzA0MjAyMDUyMjFaMFYxDzANBgNVBAMMBmNhY2VydDELMAkGA1UEBhMCRkkx
+VQQHDAhIZWxzaW5raTEQMA4GA1UECgwHTWFyaWFEQjAeFw0xODEyMTYxNzU3MDBa
+Fw0zODEyMTExNzU3MDBaMFYxDzANBgNVBAMMBmNhY2VydDELMAkGA1UEBhMCRkkx
 ETAPBgNVBAgMCEhlbHNpbmtpMREwDwYDVQQHDAhIZWxzaW5raTEQMA4GA1UECgwH
-TWFyaWFEQjCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAKCt1bHsRW/W
-M/xaAykU8Y541SdTeeCSfBA7eaDXtp2oXE36aBGzA57uXiB5I9icSTScHcRuUx+a
-kh8IwRXirc9ZzR5VhHn5Cco2ilCDxjhIxtP69vIqT71dYJ3rIcSM8t0tSRBjRkfe
-LVmgSuBY5sCu2NRemvj1aB3qgIrWAbDVXzBNiFrFH4GSwUBUyLumoUPegTxLeZWC
-u1Lao6Sgaf9+AIyGheyvA2iog0ig5B0xqVxHmZ06P7U+EnxNRxVy8RFcSu8IHHuP
-5gMGB0+UIbBeJ/qTjLTMVjQ7bcRKFFeyIRo+L8WeRxpZBSIOVrGn6ICbgsNUVxIF
-lHmiA9lkPGMCAwEAAaNQME4wHQYDVR0OBBYEFBzHK6obsbsumvQPsYZgVzjCQQUS
-MB8GA1UdIwQYMBaAFBzHK6obsbsumvQPsYZgVzjCQQUSMAwGA1UdEwQFMAMBAf8w
-DQYJKoZIhvcNAQELBQADggEBAA1LIVL6STRWFNuDrhw9p00+6lV+Gjd6ZYnuGQWU
-nTqtWcQ4FrK9Au5apn7isSGjra+MrsMwca3X0iQPxNlHgMWVBR18ikkKfYthyrVo
-PT5O8cdFYsjMqS/zEvE/kjR/B6vTrKuvLclpY4qy5TXqfbgXOHKCX5Y93I3lEbuu
-8wItIHdcZFkYpuf6x4noMBIUBEBb6bGPhoG5DWy2/Jj5t1Krj35TyKAF5M0Na9J0
-nxd6ocN2XvMpHMa+VqsC913hySEnbWZ6QSlJo/j1KucDKnxSS/VGWEW+pEygZTcd
-2Kz4H4HKnHnw/yKMHc4r0B7OmfLb+maE5oZvGTsQ8ZKsV7I=
+TWFyaWFEQjCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCCAgoCggIBAMOf6OXiujVY
+b12il3HhBCknyGfYEn/LhuTexwYhNH36H8tB9yG5qeiHl9GmktA/IJ+BsmJjWcHV
+xR+bhg48Z2LdAf3q09tVEAZaUjjquef6r3SQdWXGh99Ci8MLAg+WnVg7cd8ACVad
+01IrgPI/sc0+abnhKMvIWNnSu7j9+ZrPRii0iFyQNMKIghjs4Ebe0fGZNba9n63Q
+hz2jbOOwAfBJynFNhAerl0u/sLMXbykw2QfRpEnq2nvBUF4AVPjNeeTUITHQG7I8
+ODALZr35B7z9durP7oR6UphbnC8REaY/Y1M57cPohFbHX0duvFsbH19wwJBJXpG5
+IJB+X3g3/IRnlzH7so6EpSQ9g5+/GUtkAjrouq798gymo5/zzOm+bjAnCuierTjU
+qxADtCi6xdHAy1GL7a40HnQd4Bt3qdoCZ4pYl0fvMWYlz+/hTXJQksNlzFlJGsex
+ueaRcAWAzmGdJ6CIDaTkicPp6Wi2bBqwopo5Bvu5V08fkYy9YGEw+IjT6n1WooUJ
+cXBQ1W6JLbjZ5d9MDcq4SdxdIUzXT/Yis4Af2i2P15zPSmYeAZaVkt9OsMhH80sN
+QK+vwmb0QWOlnvIHqX1vOr4MoxVc5fByfK8op/42yesf9HN5DkSfS45mB6NiilHr
+9mZkSdlCoRnb+HllW7Tg4YBIZgnJ/vkFAgMBAAGjUzBRMB0GA1UdDgQWBBTaLKFj
+njMdRPLhNxfJj5BPafiwhTAfBgNVHSMEGDAWgBTaLKFjnjMdRPLhNxfJj5BPafiw
+hTAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4ICAQBqTxY9cHVSiHWF
+yDkpprhYgMaF4M0kTBUxWkIZz7O2Tllez/1VPzgJyaj5wRDXJWh9e9kK5XQ0A1Y3
+eQkSFJsOrxbEmT0HUzmPI3P6jT6v8EU2xqFxqQoi9/KGs0bkpllIqDb79l5sDp8N
+DsIyyRuk0wdqYQ03KJaw6FzVORbaEkfHhgUxg4PT8OqSOSJqLnPaRrH6xhmjGh7e
+Z+BXdFh6DdelfsTjaGdVAK72KIkewSWPaSS8X/z4NFrennuZPdiK2T2vpApxAGZS
+j5QIv2kJSOzx4BiE7vOfn+VRRDvmVnD1X/kluUQq3YkokMLyCjYQ1Ad5EUh/SQ1g
+CXFqaqVybRGgM0h2Devm3AHDBkj7OtRwczjdaxgTEMqr9sDB8PGbYT0bfANSTSSK
+aQcQpyG4gRji31RS95v6EOyzNQiEhEJIQuLtMI6qIPIjysH/MJEB/gekjsIwCf14
+fmjCNG3RsRJewRgc9KbbaJbXXP/Hbqfkz39D0P53fVGLSEElVYbcoZ56MmMnHiQA
+2i+gKHhVZS03bMJT4b7SVKagGyCHDL0SGF52Y3GBsfzz4PHcPa22wDIP5DXKFAxr
+WqJNSm4IsGbkJDvockfQTyMpYbYQh2xCIlu7gtY+X+ATnNY4ESOkuz/xVkR0MrW4
+2RnjcZ+YSQ2l47ZsqU/68yl1NZoEGw==
 -----END CERTIFICATE-----
--- a/mysql-test/std_data/cakey.pem
+++ b/mysql-test/std_data/cakey.pem
@@ -1,28 +1,52 @@
 -----BEGIN PRIVATE KEY-----
-MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQCgrdWx7EVv1jP8
-WgMpFPGOeNUnU3ngknwQO3mg17adqFxN+mgRswOe7l4geSPYnEk0nB3EblMfmpIf
-CMEV4q3PWc0eVYR5+QnKNopQg8Y4SMbT+vbyKk+9XWCd6yHEjPLdLUkQY0ZH3i1Z
-oErgWObArtjUXpr49Wgd6oCK1gGw1V8wTYhaxR+BksFAVMi7pqFD3oE8S3mVgrtS
-2qOkoGn/fgCMhoXsrwNoqINIoOQdMalcR5mdOj+1PhJ8TUcVcvERXErvCBx7j+YD
-BgdPlCGwXif6k4y0zFY0O23EShRXsiEaPi/FnkcaWQUiDlaxp+iAm4LDVFcSBZR5
-ogPZZDxjAgMBAAECggEAWmy6AGFpSmEP7IpzkOEaeAWEX5dY1YtaioAOGPiM6vje
-yXuMqblG5mBbVIcYJ0T85cCd9/fmi7ifVxvEHh7tle2Bw/p4jXQbkFNVT655FR/P
-1Wg9JVeufHFaeETlQgnYe6SKo9BaswNUHkZZHRyq7/D2Ub3UFRt2tq9MG9YIKY1m
-rP9s7E+EDuH9UhYmaWdQfNm8muIXWK8WjicI5+PX0CQ1NtUy6vS7qBzcBzvT0chC
-Jtja29S6Nvg12A96nHsRmQyUaQjRlqosSwiagpc5mZmNeCEUoY+3deIdYIUMSQnf
-judZOKVPq0GOW5Y1U068LGODWaifPkinGBj+04VH0QKBgQDOp/jVCOUdEeqFJ/8m
-wEsfsRIrXvtGJHgbDXcVJ69FwlX+yaKGEuC+4f21uyxPn6GoFw+NKAyTmGKH7VAX
-OFQLrMQ/DMlNbZrCAAFcXMqrnLaVwqMeIIoVNfKAa8u15K40qc+B0it61Nlay5wq
-wvXoSZrdqXSgsI29pav20+8pTQKBgQDHC3l1+gMZ1rCar+5KdVBN1Wq4Xh7cwZw6
-FxEvyrDCJePEU2L7FpH1pFuB4WpXdBu3CPo70ZgwfqBXn4qLOOI3gTtDHActyiUm
-+WRG62O+5Ye7aLB4xy0MfnKNA2g/yHj1ozwM8kA5JRptAzDnzWfVE0k47/pVAVzt
-E2bZuSykbwKBgQCL6SkMgjMr1T9j20phn/q8gBN/DZUtTe+K0Tj4N5/wqLuz/its
-fkdutG4ipZBAcCDwPnym4qBxJNBAmqiIr/gm11ceILgBFd2azoodUC1etoDfL6Fj
-+j/CUH3X+CM5CJPwz67Pg80wIf7t+7/FK611ELAqtllhmWa9KPcd6yqWWQKBgHh5
-Xnvk5kmWY3BNOgrBNOjXWu/asA1n9lpGqfVmVlQ8wL6MxiU5xQCMCYL0X/ws37WK
-boMUWmxHyF8gxqd7t5hm1OrKpSG274PGgUZXpRjfLqdlNyLzUzXztvvY6xloCqaK
-tYcUfYDZD0SaINi8v7L9KF2ZCsi2uXsZOjBf30BrAoGAXPPotkw/CkcPQBS13cha
-ZWeeH5NDKBADWXfLfcRUs108c9xw4BYr5yGilSPscN2ZP0/iWONKp/c6/STS54t5
-lkOKKUbkAFbQu8UKa1J7zrnHZv+Mr4I/iBBy6VkN8Spp2vBI3Ng6jhPIJg3Gum9p
-943wWtAnIhe/UqCRT3a/GZg=
+MIIJQwIBADANBgkqhkiG9w0BAQEFAASCCS0wggkpAgEAAoICAQDDn+jl4ro1WG9d
+opdx4QQpJ8hn2BJ/y4bk3scGITR9+h/LQfchuanoh5fRppLQPyCfgbJiY1nB1cUf
+m4YOPGdi3QH96tPbVRAGWlI46rnn+q90kHVlxoffQovDCwIPlp1YO3HfAAlWndNS
+K4DyP7HNPmm54SjLyFjZ0ru4/fmaz0YotIhckDTCiIIY7OBG3tHxmTW2vZ+t0Ic9
+o2zjsAHwScpxTYQHq5dLv7CzF28pMNkH0aRJ6tp7wVBeAFT4zXnk1CEx0BuyPDgw
+C2a9+Qe8/Xbqz+6EelKYW5wvERGmP2NTOe3D6IRWx19HbrxbGx9fcMCQSV6RuSCQ
+fl94N/yEZ5cx+7KOhKUkPYOfvxlLZAI66Lqu/fIMpqOf88zpvm4wJwronq041KsQ
+A7QousXRwMtRi+2uNB50HeAbd6naAmeKWJdH7zFmJc/v4U1yUJLDZcxZSRrHsbnm
+kXAFgM5hnSegiA2k5InD6elotmwasKKaOQb7uVdPH5GMvWBhMPiI0+p9VqKFCXFw
+UNVuiS242eXfTA3KuEncXSFM10/2IrOAH9otj9ecz0pmHgGWlZLfTrDIR/NLDUCv
+r8Jm9EFjpZ7yB6l9bzq+DKMVXOXwcnyvKKf+NsnrH/RzeQ5En0uOZgejYopR6/Zm
+ZEnZQqEZ2/h5ZVu04OGASGYJyf75BQIDAQABAoICAQCm3h3NEpSh8ni0l3+1rSpe
+bMf/BInrJtI/ImUErohXOWereu5yfywW5Kq1M/Ch4GdUiE6ArJygwlduubwlIpNR
++pG7ZXcn7axIrvzZ/idsYSMPPDZYBXxbDimnyxkL8P7wh2x/u0k78E6VFfTdL0Xc
+fHO1EgI5qgFYdPCaQVKMImiHSk/THVNT6RuIZW5+JY1SF82AvsmZe3oRMvksUzOk
+I12jPmr0HANnovD4I3I34aoPiF0qjEGoNOL3/DJ/NM7YSbazBgDKDaIkIWWXgPh/
+1nl8nLepm6oeum5Em7vBYuq30H8nkIxqdZCpnyTeVglx4PzxVxRrG05S0ule4+rq
+DElymQHcUEv+zRAD+9zCWXLduDiGk1mLbWNc1sYRC7Q8mqRWxf9XZFMG8p2L6ws8
+RRqVnXkfU6ZDswkW27vwX4IWiothPh2Hli92vuzNqxf/XMHctex4q3qFBApj2xdz
+VDj+lZWX99DOJNAtbmf2r4CWvGPDwyQRNSa4VwXAgou5klAlbb0vyg6fco7Ybqyz
+A5ueZZCGLQ5wunEDEUXyVVSTuG93tTOQ6cqoQbApcdifLA228iaOTDw8jnjKtufV
+6F41QXC9TKUy5DCM3iyPlobbmQdlQwylDw3oAjr3pkLH/SamUQ4REh16xCZ6h+XY
+Ore3SEtmvaqUKYymHo3HwQKCAQEA6cW7vuSUGwWYDFS65Si3pcidx5BOxWLvQXRV
+de/CFBkjnsK+qynEXTSw61qt0j2ACv2JRGFogyZFoTPvIueeFWmIKYQh8gB1DB/x
+jvo+IPjTdZNEIJywJMzp1jxhHkO5iovXUXr8/xu7jy3dFXscg8dgUt31iRmhPRWh
+DidkE9huKvoU0P6cLTYb4LVraVboUb+BBhjCvmQftUqPUHYdQmOwyP24HRgpxmoX
+1VeBkT9YBdj/qKDJe2hWBzcGjk0pMHrvoP26Rc5ZQBqf8eZfHE3blQrgAlIlRlko
+D0I8IWhSLCujVGa1OXZN1XjazE4gQaUUF2NEXv6qWviYlR41dQKCAQEA1jmetTP7
+Yfg+rCT5j0wfRUmouop6sLGFROo5WblYXQNUL6hkhQuIj/p2PafDB9KTn7o3TR4d
+20HUcRCZWkz5d9A80wQqa2mwyy7qUl0cYgWIJfXzn97+PZeOlyo1iONGPsxCaGsc
+TA8UbDvDBvC0Njqi1KRTzMTS4DMFAULtlp+5zmd1iFNWgeNj2xomkiFU1AI4ALHW
+4fxZzltlwxL2d37ENh4tMSgKmvSHLR4b1xFyvQpFaRmxSaBLlAN5WRaMA9scvRMI
+X1yGF/1elib5ZcvnDZuiCJa/RVc/5BbE1dGMcpOPzqrbvZDwefcmk1+1qlqRJuMX
+IIlcPVC406rzUQKCAQBEIViiPcPcTDkP/vGpL8C2FE3Hh4stg7PvA5SUlpede+Yh
+sugWwkl0UJLvRLI7/6h9i36h6zGmZ1ULsjMTnSU5YFRKpTOvb+qs6fJglRXeTiZy
+scS9LG05jG7p/Rbg1yQP+Ml7+bgaMV7sT5WLWSHqVgUAoaeZmVqgjtktYtWL4xvY
+cqH2dv3VP1vIlZcEUEvM518XA8hd7JhBIcW60XGdk6hGK4upDw29wdXIW4RrbGae
+7as03gdabAIGgEbs2hgTBQEpjkjnZ0gjop7SxS0a9qX0PgMjKgWrBvjh2Y/kvJ6W
+OPGU+4n9gTkV6e76dBp6hz3fwxJ+DQJhGOp/KxqxAoIBAQC5RF+zSl9vXOemrNt/
+SFAK9icMHNL6mKwu63np1NdMqaJXDQTR4yWZjvxPWhTbYQv2qW9GLHOZuZbmqXpW
+lBdzvQ29WO7ushsJvWs9C/WN+EbEva6SZqHt1nKk3QYrepK3oKMV9vwj8y6wZAbI
+q8bO12sXB37JqrbM2pLLgnxDNaPVk7gbzCcWcfdb94PAAT+BkuGoCCVUj4+DXZvm
+VzH8Kg+61eIzjvONC0kUheOB3j8iqg1ZVbTzfcOfIFoD5/HzK73w7tLg1f3KJX91
+0AMEPJF4PU6mquiEKW7B7ccRPAaA2IkYLL+ZdA3lDQ+5iJ5t06GQxfisVN8G0Bbo
+SAHhAoIBAFV3f55jwEA/jWMaZNRtj2aqQBhFugOsMcpa417nJKtQkgYhn3qviFTD
+X12G24mcURjpIBmFsaPjGVybGDGuBGbizmkQ1W/xQ8e57nuMoazwBJpYIFrvnwde
+EqRv0hYYipy+x6O0KMHXrju9osWyStx3uPysz9uCuCEiEI4vK9CYCbp1mx3VFGNj
+cstymyn0L1TKoH1M8ADyhOTBuRzad471f1yxjczh4w9OCg0lz51o1/UJln4UvvWV
+HMG9KwjHJCbrn9WreVkm6Ko0AAFg51CyEh9n2KL2Ze1uUEywQWeMj65i2a4y3s/9
+WPsvqYkXIuKwfPNClXeg/lB3ng9YG5o=
 -----END PRIVATE KEY-----
--- a/mysql-test/std_data/client-cert.pem
+++ b/mysql-test/std_data/client-cert.pem
@@ -5,22 +5,48 @@ Certificate:
     Signature Algorithm: sha256WithRSAEncryption
         Issuer: CN=cacert, C=FI, ST=Helsinki, L=Helsinki, O=MariaDB
         Validity
-            Not Before: Apr 25 20:52:33 2017 GMT
-            Not After : Apr 20 20:52:33 2037 GMT
+            Not Before: Dec 16 17:57:11 2018 GMT
+            Not After : Dec 11 17:57:11 2038 GMT
         Subject: C=FI, ST=Helsinki, L=Helsinki, O=MariaDB, CN=client
         Subject Public Key Info:
             Public Key Algorithm: rsaEncryption
-                Public-Key: (1024 bit)
+                Public-Key: (4096 bit)
                 Modulus:
-                    00:a1:10:ea:cc:8e:2c:73:6b:33:1a:5e:26:19:b6:
-                    4b:4c:bc:04:b8:c2:e2:33:eb:67:a2:7a:27:af:3f:
-                    f7:ef:49:5f:c1:d2:b9:d9:71:fe:17:a0:93:da:dc:
-                    f1:47:de:fa:1f:c3:c1:d1:a5:2a:06:cb:b3:e8:9a:
-                    c1:bd:78:77:68:45:c1:55:cd:b1:c1:d3:df:8c:12:
-                    4f:c2:3a:0d:b7:58:dc:ca:13:08:b9:fb:12:24:90:
-                    aa:b7:4e:04:eb:43:0d:45:be:1c:17:d6:a8:b1:af:
-                    10:3c:39:d6:08:45:ed:a9:7e:3a:69:ae:70:22:86:
-                    7e:71:1f:f1:0e:d0:0d:32:c3
+                    00:f1:52:d9:f1:37:47:09:3d:b1:81:ff:27:46:33:
+                    a0:68:60:e8:b2:1e:6e:9e:14:c4:71:a8:6c:8f:15:
+                    69:cd:14:78:2a:23:ad:8d:ba:23:21:f5:94:e5:43:
+                    09:14:43:bf:ad:98:e6:91:e3:5b:b3:98:90:85:c6:
+                    32:e1:25:34:4a:bd:ed:0d:75:42:c9:7e:e1:f0:30:
+                    19:75:83:fe:0b:1b:34:46:dc:6b:49:5d:d2:74:95:
+                    32:14:b0:fb:cb:2d:74:7e:1d:12:05:80:09:bd:6c:
+                    15:98:3d:8e:08:bc:1c:a1:c1:e0:c9:22:a5:d0:eb:
+                    56:12:4d:c4:5f:75:87:6a:48:8a:61:77:20:b4:bb:
+                    3c:56:79:b0:1f:fb:f7:c8:dd:a4:56:95:72:6b:5e:
+                    55:d1:80:ad:60:51:92:11:64:c2:0c:e0:e8:b5:9b:
+                    51:1b:32:38:c9:28:e3:d2:8e:26:a3:f0:3e:a6:11:
+                    e6:fa:c4:26:95:63:34:6a:5b:ad:ba:95:91:29:36:
+                    79:d2:5b:0c:e9:db:bb:50:52:c6:14:fe:e3:92:93:
+                    09:e6:07:0c:6c:e6:7f:08:f6:69:ea:70:5a:34:8e:
+                    ed:97:23:ad:e4:5c:0f:3b:c3:fc:54:d3:86:42:e0:
+                    1e:1c:95:31:41:cc:3e:5b:de:6d:b6:a3:41:91:ed:
+                    0c:18:3c:03:00:28:1a:f1:29:fd:d1:d0:cd:95:9a:
+                    f6:95:05:05:af:dc:f4:e8:85:88:9e:e9:81:8b:64:
+                    64:e5:7a:2c:77:1c:ea:c9:06:9c:78:92:07:5a:03:
+                    4a:e6:f1:de:b3:0d:98:36:fc:8d:c1:9a:00:58:26:
+                    64:d8:42:a6:d5:d4:ca:a0:50:fc:90:d9:42:a9:95:
+                    ac:be:eb:7c:a2:e8:b2:62:d8:ef:ff:9b:95:37:cc:
+                    a5:34:2c:7b:2b:79:a2:42:26:0a:1c:72:e6:05:3c:
+                    22:f4:77:81:e7:be:d7:9a:7e:3b:94:36:36:c3:58:
+                    99:8e:17:11:63:c7:44:fe:44:32:4b:0b:24:e3:c8:
+                    0b:60:8d:fd:37:e7:86:f9:98:ae:a7:f6:0e:d9:f2:
+                    bd:92:e5:4f:e1:76:50:05:a5:15:ea:ec:96:f4:61:
+                    95:56:72:41:14:eb:cf:8e:3d:f3:18:38:8a:6e:71:
+                    fa:36:6e:1c:80:93:82:f3:7f:bd:81:ad:c9:8d:cc:
+                    12:b9:d4:8e:45:d0:98:90:36:1a:d7:48:32:66:6f:
+                    08:26:a2:39:b8:17:86:63:b3:22:16:87:6a:4f:f8:
+                    1c:c3:aa:cf:7c:9d:bd:37:ba:bf:dc:d9:a4:05:fd:
+                    3b:6b:1a:4e:42:eb:5f:dc:c6:a0:7e:27:25:82:39:
+                    38:d2:35
                 Exponent: 65537 (0x10001)
         X509v3 extensions:
             X509v3 Basic Constraints: 
@@ -28,42 +54,70 @@ Certificate:
             Netscape Comment: 
                 OpenSSL Generated Certificate
             X509v3 Subject Key Identifier: 
-                0C:20:76:A1:80:9C:2F:30:3D:F7:AB:8D:31:19:AD:E2:F7:E2:8D:12
+                32:09:91:7E:CE:1B:13:63:3D:E6:9B:66:17:40:B1:56:7C:4A:78:D2
             X509v3 Authority Key Identifier: 
-                keyid:1C:C7:2B:AA:1B:B1:BB:2E:9A:F4:0F:B1:86:60:57:38:C2:41:05:12
+                keyid:DA:2C:A1:63:9E:33:1D:44:F2:E1:37:17:C9:8F:90:4F:69:F8:B0:85
 
     Signature Algorithm: sha256WithRSAEncryption
-         39:c0:90:13:19:85:47:9d:c6:ab:8c:c6:c9:0f:33:11:19:f7:
-         01:2c:1b:08:f6:81:98:11:ab:48:05:d9:b2:29:56:32:9c:ba:
-         e5:40:df:85:5e:6d:fd:6e:36:9a:14:eb:90:50:57:de:2f:ed:
-         2d:89:a6:8a:40:1c:41:84:9b:da:e1:6d:e6:7c:46:b2:e0:90:
-         93:02:1c:52:2e:af:b4:d4:a1:d8:9d:19:cf:0a:67:bf:c3:3e:
-         2e:02:f4:3e:bc:2e:59:57:30:85:8a:32:ab:22:88:72:37:6e:
-         ee:ed:f8:53:72:c9:28:87:50:47:81:1b:80:4c:f8:80:ce:2f:
-         47:ca:78:ce:38:51:70:ec:df:ee:fc:ea:5a:40:1e:4d:1c:fd:
-         4e:f6:74:d0:22:a4:7e:57:df:16:1a:a0:8d:be:fe:ee:f2:07:
-         2e:39:a1:97:40:19:f9:3b:b8:e7:c4:98:6e:1d:1a:27:d3:19:
-         4c:5c:c9:c3:31:98:c1:3c:27:0e:6a:de:cf:88:72:cf:e4:65:
-         c9:0d:33:32:f1:ea:f7:dd:5b:9d:42:6d:ee:c7:a8:b7:85:d2:
-         41:e0:84:38:ce:86:81:ba:6e:7d:d5:ad:7a:00:58:d7:c5:83:
-         9e:5c:1d:38:32:72:49:f5:42:4b:e7:c6:5c:12:6d:e1:5d:51:
-         2c:f5:52:f0
+         6f:7c:37:c5:27:b6:21:79:60:32:6c:41:b0:15:1f:76:98:7d:
+         dd:44:ad:26:1b:5a:d3:5e:4e:25:df:5c:47:2b:da:c1:e1:55:
+         47:9e:ac:44:a5:78:a3:2a:da:b9:25:7f:21:f5:dd:50:c1:b2:
+         58:32:30:15:be:15:5d:3a:af:ba:f6:06:bc:f7:4a:b4:a9:41:
+         32:c8:05:e3:7b:e0:b7:e4:03:dc:57:8a:32:df:e7:95:a0:f3:
+         e6:24:e7:d1:b1:4e:22:1f:ae:ef:1a:2a:69:be:8a:97:30:af:
+         f2:b0:9f:ec:29:71:d4:55:09:d0:dd:0a:df:26:d1:34:f9:df:
+         e7:4d:5b:2d:1d:6e:52:a5:85:94:6a:bc:46:64:04:f1:d7:f1:
+         cc:38:e4:27:ca:bb:ee:65:dd:eb:0a:58:45:ad:0a:27:04:43:
+         4b:0c:db:64:2b:23:51:e0:1b:43:c1:62:d4:48:df:9f:39:e9:
+         e0:dd:b9:b7:ca:34:c0:07:4d:42:e4:0e:04:7c:70:7f:62:0f:
+         69:4a:47:be:b1:67:b7:16:e8:c1:f0:80:6f:8b:87:60:4d:01:
+         3f:54:f2:1c:46:69:11:2b:97:53:e8:5e:d9:66:62:06:5b:19:
+         d7:fd:14:90:f5:3d:ea:97:5a:45:0f:c4:f8:ee:fa:6d:fa:1c:
+         42:0a:39:96:a0:87:11:ec:15:3c:ab:8f:f8:05:c8:40:3f:c7:
+         b1:e7:bb:73:9f:51:8c:02:89:b8:45:d6:99:42:c6:4d:23:e3:
+         6e:5a:46:19:68:26:9d:0d:9c:65:67:13:1e:60:c4:75:af:b5:
+         c8:35:5b:b1:c3:19:0c:b6:80:20:11:cd:c2:4d:87:96:3c:62:
+         00:cb:03:65:5e:62:1a:58:5d:d1:55:da:b9:34:06:75:f0:54:
+         86:89:94:0a:d4:84:c7:0c:b7:e1:70:6e:f8:b0:30:11:99:47:
+         bc:a6:79:63:b9:2a:a1:8b:84:29:6f:56:34:b5:1d:e3:09:9f:
+         26:8e:eb:98:dd:f6:78:ca:5c:63:c2:99:80:22:4f:c1:77:de:
+         ea:36:51:a9:3d:3e:42:ea:7e:23:b5:b4:62:4a:6f:55:39:66:
+         4e:23:2a:84:d5:bf:51:06:c3:57:56:7e:cf:b3:80:fd:e4:b8:
+         07:76:30:97:28:32:3d:c3:a2:2c:de:0e:07:ed:7a:44:69:21:
+         2b:9c:ea:0e:cb:72:6a:72:33:66:dd:2b:ae:71:00:71:38:40:
+         e4:3d:e4:fa:63:ac:04:45:aa:a1:bd:31:1f:13:45:30:6a:e2:
+         1d:be:d3:30:d8:4c:e7:61:40:88:e8:14:90:42:41:63:71:18:
+         69:e5:f9:66:d0:ff:a7:2c
 -----BEGIN CERTIFICATE-----
-MIIDHjCCAgagAwIBAgIBAzANBgkqhkiG9w0BAQsFADBWMQ8wDQYDVQQDDAZjYWNl
+MIIFojCCA4qgAwIBAgIBAzANBgkqhkiG9w0BAQsFADBWMQ8wDQYDVQQDDAZjYWNl
 cnQxCzAJBgNVBAYTAkZJMREwDwYDVQQIDAhIZWxzaW5raTERMA8GA1UEBwwISGVs
-c2lua2kxEDAOBgNVBAoMB01hcmlhREIwHhcNMTcwNDI1MjA1MjMzWhcNMzcwNDIw
-MjA1MjMzWjBWMQswCQYDVQQGEwJGSTERMA8GA1UECAwISGVsc2lua2kxETAPBgNV
+c2lua2kxEDAOBgNVBAoMB01hcmlhREIwHhcNMTgxMjE2MTc1NzExWhcNMzgxMjEx
+MTc1NzExWjBWMQswCQYDVQQGEwJGSTERMA8GA1UECAwISGVsc2lua2kxETAPBgNV
 BAcMCEhlbHNpbmtpMRAwDgYDVQQKDAdNYXJpYURCMQ8wDQYDVQQDDAZjbGllbnQw
-gZ8wDQYJKoZIhvcNAQEBBQADgY0AMIGJAoGBAKEQ6syOLHNrMxpeJhm2S0y8BLjC
-4jPrZ6J6J68/9+9JX8HSudlx/hegk9rc8Ufe+h/DwdGlKgbLs+iawb14d2hFwVXN
-scHT34wST8I6DbdY3MoTCLn7EiSQqrdOBOtDDUW+HBfWqLGvEDw51ghF7al+Ommu
-cCKGfnEf8Q7QDTLDAgMBAAGjezB5MAkGA1UdEwQCMAAwLAYJYIZIAYb4QgENBB8W
-HU9wZW5TU0wgR2VuZXJhdGVkIENlcnRpZmljYXRlMB0GA1UdDgQWBBQMIHahgJwv
-MD33q40xGa3i9+KNEjAfBgNVHSMEGDAWgBQcxyuqG7G7Lpr0D7GGYFc4wkEFEjAN
-BgkqhkiG9w0BAQsFAAOCAQEAOcCQExmFR53Gq4zGyQ8zERn3ASwbCPaBmBGrSAXZ
-silWMpy65UDfhV5t/W42mhTrkFBX3i/tLYmmikAcQYSb2uFt5nxGsuCQkwIcUi6v
-tNSh2J0Zzwpnv8M+LgL0PrwuWVcwhYoyqyKIcjdu7u34U3LJKIdQR4EbgEz4gM4v
-R8p4zjhRcOzf7vzqWkAeTRz9TvZ00CKkflffFhqgjb7+7vIHLjmhl0AZ+Tu458SY
-bh0aJ9MZTFzJwzGYwTwnDmrez4hyz+RlyQ0zMvHq991bnUJt7seot4XSQeCEOM6G
-gbpufdWtegBY18WDnlwdODJySfVCS+fGXBJt4V1RLPVS8A==
+ggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQDxUtnxN0cJPbGB/ydGM6Bo
+YOiyHm6eFMRxqGyPFWnNFHgqI62NuiMh9ZTlQwkUQ7+tmOaR41uzmJCFxjLhJTRK
+ve0NdULJfuHwMBl1g/4LGzRG3GtJXdJ0lTIUsPvLLXR+HRIFgAm9bBWYPY4IvByh
+weDJIqXQ61YSTcRfdYdqSIphdyC0uzxWebAf+/fI3aRWlXJrXlXRgK1gUZIRZMIM
+4Oi1m1EbMjjJKOPSjiaj8D6mEeb6xCaVYzRqW626lZEpNnnSWwzp27tQUsYU/uOS
+kwnmBwxs5n8I9mnqcFo0ju2XI63kXA87w/xU04ZC4B4clTFBzD5b3m22o0GR7QwY
+PAMAKBrxKf3R0M2VmvaVBQWv3PTohYie6YGLZGTleix3HOrJBpx4kgdaA0rm8d6z
+DZg2/I3BmgBYJmTYQqbV1MqgUPyQ2UKplay+63yi6LJi2O//m5U3zKU0LHsreaJC
+JgoccuYFPCL0d4HnvteafjuUNjbDWJmOFxFjx0T+RDJLCyTjyAtgjf0354b5mK6n
+9g7Z8r2S5U/hdlAFpRXq7Jb0YZVWckEU68+OPfMYOIpucfo2bhyAk4Lzf72BrcmN
+zBK51I5F0JiQNhrXSDJmbwgmojm4F4ZjsyIWh2pP+BzDqs98nb03ur/c2aQF/Ttr
+Gk5C61/cxqB+JyWCOTjSNQIDAQABo3sweTAJBgNVHRMEAjAAMCwGCWCGSAGG+EIB
+DQQfFh1PcGVuU1NMIEdlbmVyYXRlZCBDZXJ0aWZpY2F0ZTAdBgNVHQ4EFgQUMgmR
+fs4bE2M95ptmF0CxVnxKeNIwHwYDVR0jBBgwFoAU2iyhY54zHUTy4TcXyY+QT2n4
+sIUwDQYJKoZIhvcNAQELBQADggIBAG98N8UntiF5YDJsQbAVH3aYfd1ErSYbWtNe
+TiXfXEcr2sHhVUeerESleKMq2rklfyH13VDBslgyMBW+FV06r7r2Brz3SrSpQTLI
+BeN74LfkA9xXijLf55Wg8+Yk59GxTiIfru8aKmm+ipcwr/Kwn+wpcdRVCdDdCt8m
+0TT53+dNWy0dblKlhZRqvEZkBPHX8cw45CfKu+5l3esKWEWtCicEQ0sM22QrI1Hg
+G0PBYtRI35856eDdubfKNMAHTULkDgR8cH9iD2lKR76xZ7cW6MHwgG+Lh2BNAT9U
+8hxGaRErl1PoXtlmYgZbGdf9FJD1PeqXWkUPxPju+m36HEIKOZaghxHsFTyrj/gF
+yEA/x7Hnu3OfUYwCibhF1plCxk0j425aRhloJp0NnGVnEx5gxHWvtcg1W7HDGQy2
+gCARzcJNh5Y8YgDLA2VeYhpYXdFV2rk0BnXwVIaJlArUhMcMt+FwbviwMBGZR7ym
+eWO5KqGLhClvVjS1HeMJnyaO65jd9njKXGPCmYAiT8F33uo2Uak9PkLqfiO1tGJK
+b1U5Zk4jKoTVv1EGw1dWfs+zgP3kuAd2MJcoMj3DoizeDgftekRpISuc6g7Lcmpy
+M2bdK65xAHE4QOQ95PpjrARFqqG9MR8TRTBq4h2+0zDYTOdhQIjoFJBCQWNxGGnl
++WbQ/6cs
 -----END CERTIFICATE-----
--- a/mysql-test/std_data/client-key.pem
+++ b/mysql-test/std_data/client-key.pem
@@ -1,15 +1,51 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIICXQIBAAKBgQChEOrMjixzazMaXiYZtktMvAS4wuIz62eieievP/fvSV/B0rnZ
-cf4XoJPa3PFH3vofw8HRpSoGy7PomsG9eHdoRcFVzbHB09+MEk/COg23WNzKEwi5
-+xIkkKq3TgTrQw1FvhwX1qixrxA8OdYIRe2pfjpprnAihn5xH/EO0A0ywwIDAQAB
-AoGAaJMqT+vTcpDcmjcct78DPuwoiKmH4rvqCrUTRRbcbJGQSbD/F+6KUl7hAM5J
-Hifo8GzST8LT+ZuS0OiB9/naVGJjcLzpV+us+keMctB/cjmsPAAsRdeA8Xk0jTWv
-v+5a5ZWSSbRXycuUtywtoESY1RLEyB0k3Dcxvk6SD3LnNMkCQQDTyDJfP+eirhgC
-mKiNvHJjLtlRT11IMLMVTELEntsyKqzVgficZCWGkFRoF9iEO1OyCJaug7RZsuxs
-Z3lk/gq1AkEAwrHwzVPU+dTqsZ8tYHlq+d0xy+6eTtYy7e/5qH3AUz10us6BG/LY
-XVTwRFAOKQOciKZ/zPQa4oYeAc0oozSalwJBAIuMbyS0Rz262bdcQDSk5/rS8//P
-+/eFn3t5NMW6p1T3KcvGSLtEgjWZBQVSMSlwrkWxwxhbUIuKip8jz6nse8ECQDgm
-g5FkLRdEfc9uXfLl8aFQVu0+y29nPVb8D9+1LMOVBNZfekLqPdZlCcpZ4EuZ3ApZ
-IqCkgiB0l7DjT2YKZM8CQQCp8z1JvoNRwhIuojh7JajY5QBkXtvHootXr+vDUHNf
-SgR5BRmqZb0F5/BK4/7JbumiHI11QUULxhOnNtlS7DDt
+MIIJJwIBAAKCAgEA8VLZ8TdHCT2xgf8nRjOgaGDosh5unhTEcahsjxVpzRR4KiOt
+jbojIfWU5UMJFEO/rZjmkeNbs5iQhcYy4SU0Sr3tDXVCyX7h8DAZdYP+Cxs0Rtxr
+SV3SdJUyFLD7yy10fh0SBYAJvWwVmD2OCLwcocHgySKl0OtWEk3EX3WHakiKYXcg
+tLs8VnmwH/v3yN2kVpVya15V0YCtYFGSEWTCDODotZtRGzI4ySjj0o4mo/A+phHm
++sQmlWM0alutupWRKTZ50lsM6du7UFLGFP7jkpMJ5gcMbOZ/CPZp6nBaNI7tlyOt
+5FwPO8P8VNOGQuAeHJUxQcw+W95ttqNBke0MGDwDACga8Sn90dDNlZr2lQUFr9z0
+6IWInumBi2Rk5XosdxzqyQaceJIHWgNK5vHesw2YNvyNwZoAWCZk2EKm1dTKoFD8
+kNlCqZWsvut8ouiyYtjv/5uVN8ylNCx7K3miQiYKHHLmBTwi9HeB577Xmn47lDY2
+w1iZjhcRY8dE/kQySwsk48gLYI39N+eG+Ziup/YO2fK9kuVP4XZQBaUV6uyW9GGV
+VnJBFOvPjj3zGDiKbnH6Nm4cgJOC83+9ga3JjcwSudSORdCYkDYa10gyZm8IJqI5
+uBeGY7MiFodqT/gcw6rPfJ29N7q/3NmkBf07axpOQutf3Magficlgjk40jUCAwEA
+AQKCAgAqshjjJ3+G8AZW1JjTP9EShorE8MNrbosu/LhwPD6izQtRkUxkddk/ad61
+DT4u+qmQAixRBWO1C7oOIPnhgZ6V1rJ5HhT551v8DdaDRx4MQvzlyJumuFXC0J6y
+sHlsAhfJNEV6+aM/gz7y13korHd4Wv/1Ng8WPnzU8jNbHEUgFcN96QC9WR8cwZza
+m87EGAq6UuiSfHc8gMKr9uKb9zKp6qThdKBN6n3HgVeFaicF/DxpT2OJXGjjJ1Ef
+vCapi0rKetvkzr0jJwAl/u8SyeHjjeZDb13QaptxL3+fvc8H233HrsVWUL9VOKCN
+NXtcm6BYOEElvggOYvUQSiQQuFraGs7vRV+Gp5RfzL55HGA0bg3Tevm99ZSlLrrR
+V01J+5nmgnTPYiravP2AJAFRxwDxdnAIDLim5tBDFJkyLo+wiKQMHRczmKXBj9ap
+KryhnKU9YQE8imhhFhZruLvaf4MEsgKK2g0XXr5H7Nei2/XEYsFB9xoBlr7/jBQC
+Cxp1M0zTqJ2WP007j56PuPejnJUKIwwE04ybAMOKQfC2Yi3U2W2C24lFR5MN1oO+
+ZVtt1U58nHNUPgRHQQW94Of0yU54W4yMzixI1+gVl3E68PhDhYVWdWvx3tLAofA7
+ycIxZjSvx+v+tm7p5DtKQ/WxThoOl571zJFOt7NACfwmhtM8ZQKCAQEA/+y+ya98
+ewFtDLBrnvhhPDhRn3uklGgpdY3lQE3y8ZfsCt//UGy5Z5+Ut56k8jTo3DbhtMD0
+Q3zJZ9PY4bv9vcjRZv7kSzV6oyZiPLncwJWjQP26DwdC3+XHzMr0KdkZ09KTNo+M
+DDr1WVlEI8gBELEY8A6RTVqOJOfe+Kwwwu22qBX4E56ilqhgz8tWZCTJ7mQPjgUS
+ErWbZsGW77I8jaRAL5IjJ8zqsylEOII4hVcPyB1/olV8O6/ZsApSA/NwinwcoHa4
+Qj/9pzi10d4nlRHIOvWuknQMMmnHC5oj0ssaJmkkCj+OSwMbEjxovLjq/x5p9m9u
+rzUoDi4/Cok5jwKCAQEA8WUB7jist2WtciFoAZJluF7D5NLcf0E019AOkc5H5C+6
+/rdmqwgYivMES+ItopvNyITCIgIbnjL5Nb1soNZgEF7lhPPsGZ+2sd/Hnz5+gYLn
+rsMOS2FvhCxAnF6QkSBt3iUl00R+AP9GS839ziL/qWz4AnFkegXgU+9+bINhQm6S
+5d4sqYHlchttuPbRMJgdN3awLHovf9zRV+Laedxtf6Np7Bv0U5CXyMParV+rpRA4
+/7+wCEh1BmY2QZWbmW0pHKcv684Y9JIqDI7XmvPn/9onJXDfREeVxV1rXnL/bZR4
+0hlSfr7O7Crq59bGCRW0VDQTyMelERVeU934pVzt+wKCAQBpq6MLO742O9x/BMkr
+FntvlgF9zFukO24C79fcuVK6lnQL+Xk+B+Z19to9BvAsYdRUOQTw0UsKIpg2mq5O
+ZoTau/RahpxUwXwlqU+eq0+O5TpHN3x/gAwIzqO014gBT3buk1/dg1/cHOv5PHZ6
+H7bw4W7ej+VG/0D51BOosdBgbC6jMI8B6tpPQq7WnarKJraKDB6wET/pIbtMrlpY
+FkwSX+NwwPJrGJ0/5KYafwo/ZDdL9YIh0RI+wtkratd6ciXhyGU5jEuAb1OYnD3W
+rdfCIVl/nrm6FtiCaFKbY8Wu39WoTnlnEd768glXekH0ypWdST2PV2toRJaJZGb7
+PrXZAoIBAF3MvBVPXH3fOqiQppOFyZT87e3qqdehBOGfbSnnGDRo/6aeqXxoJvFH
+QTIjb0wycqIqRC1m2Eh+0iu5sY5F/woddd5V0jaawykXJcCRTiMRNOnJzbIo6GZZ
+s7mPXWmnSkn58VrI0JWenDg0kR0aqHGzcCOy4DIyOlsjQuLzokVoJhjAXmZ3nSrb
+0aT/84WIdkHpEsgVqoNq0MXv1qlnp/zjCPPAYQ5frztKL50TAeWgCiYG3GY7kY2t
+BgQyGEZkDoedx5c/fJ78dO5wSwNhJ5NZt8BqvbcQgfgxNtl6ohp4iQJK6K1rg4ht
+nN39rPI2xt25Zr6sJU4gGYlJRacsFyUCggEAHDKB9gqssZXaxOAM96tdq56792RW
+xiLCZv+Qq4Qbqo91Yx701edfMNkoY579Ky4LJu1nGcvVOiJ4LKdtvEk/NoTgfBKp
+LC/yA0sjh7EztTzxlhxlvq48mHBzXvKXgL5GQgM0988UxDCDWAnV+jOJOuFAahuu
+ic7AS8CHipr2qxzH13XZm+Wx2KvTWDShVvqENwjqQCSK9yD4iO8WKPGpauzNAIII
+JEKJ0vF8jRBYcHk6C8YxRASVhIJpROMC+GnHazrvbxsrpJt6PFdajjjLbeeba8Yl
+Qd2cLLUNvcsv/PyQluVe8+S7ldS1uX0sBtkt8kQ9zoKbs/SHddp2V/oclw==
 -----END RSA PRIVATE KEY-----
--- a/mysql-test/std_data/server-cert.pem
+++ b/mysql-test/std_data/server-cert.pem
@@ -5,22 +5,48 @@ Certificate:
     Signature Algorithm: sha256WithRSAEncryption
         Issuer: CN=cacert, C=FI, ST=Helsinki, L=Helsinki, O=MariaDB
         Validity
-            Not Before: Apr 25 20:52:21 2017 GMT
-            Not After : Apr 20 20:52:21 2037 GMT
+            Not Before: Dec 16 17:57:02 2018 GMT
+            Not After : Dec 11 17:57:02 2038 GMT
         Subject: C=FI, ST=state or province within country, in other certificates in this file it is the same as L, L=location, usually an address but often ambiguously used, O=organization name, typically a company name, OU=organizational unit name, a division name within an organization, CN=localhost
         Subject Public Key Info:
             Public Key Algorithm: rsaEncryption
-                Public-Key: (1024 bit)
+                Public-Key: (4096 bit)
                 Modulus:
-                    00:c9:fe:83:3f:0b:38:89:1d:43:15:93:5c:26:b9:
-                    80:65:41:bd:2f:63:66:5e:db:fa:33:20:d9:c7:e1:
-                    35:f3:14:3c:c8:b4:f7:09:d8:f5:b5:44:8f:6b:7e:
-                    a0:a4:3b:45:5f:e6:f4:0f:08:67:f8:5a:4c:49:e4:
-                    e5:39:31:69:8c:cf:25:78:93:a6:7f:58:e9:90:9c:
-                    61:cb:2e:14:b1:57:b2:15:9c:ea:8d:6f:96:20:fe:
-                    29:ed:2c:71:b8:4f:1f:e0:05:6c:04:b1:7e:e0:bc:
-                    42:8e:bf:95:5e:a9:5e:c9:c9:a4:64:c2:1f:59:94:
-                    14:c2:06:44:79:bc:d2:65:2d
+                    00:b5:9a:5f:f5:4e:a9:fd:48:be:b4:f1:1e:7b:06:
+                    7f:3a:30:db:6e:cd:78:98:0e:0d:3f:93:61:6a:a6:
+                    cd:3d:cb:0f:e0:ac:e0:c0:70:5e:c9:6e:b4:ac:d7:
+                    50:3f:8c:7c:6a:bd:f5:00:be:04:89:35:6f:21:08:
+                    51:33:d1:af:a6:3d:3c:57:8b:dd:04:0c:55:72:34:
+                    4e:48:8b:8f:b3:f1:d1:e1:01:2f:6f:a8:37:7c:84:
+                    c8:67:82:f6:d7:8b:0e:10:a7:a5:56:aa:fe:9c:2f:
+                    eb:50:be:75:36:f8:0f:0e:85:fa:c2:8a:b6:19:7c:
+                    d6:38:2a:9b:57:c8:5d:23:9d:0d:3f:b9:15:5a:fd:
+                    3d:5d:97:c7:f6:f0:f9:99:5c:94:f1:1f:93:d2:53:
+                    2b:d7:3c:79:07:76:0d:4c:19:c8:5c:85:5f:f2:e8:
+                    eb:19:d9:0f:f1:75:10:d2:80:fe:7c:f4:d0:67:f9:
+                    8c:30:1d:46:cb:b6:90:cc:95:e3:af:02:7c:75:3a:
+                    5b:89:94:99:0e:ee:c1:e3:25:d5:26:e6:72:b3:51:
+                    a4:f6:a1:a2:b9:d1:66:13:cc:94:cd:29:27:ca:43:
+                    ae:3a:37:9d:a8:a9:a0:c1:d3:5b:3e:5e:17:4e:43:
+                    b5:c6:61:5a:55:2f:9d:e2:e1:2e:50:62:0d:81:3c:
+                    c5:bf:55:f6:0e:83:4e:6e:b9:00:40:6e:dd:ec:75:
+                    d7:96:eb:24:d1:9b:f0:ff:42:b5:d8:7c:cd:d4:fb:
+                    48:7a:a1:0c:e3:b0:88:bc:a1:4c:8a:56:97:d8:87:
+                    47:1a:47:fe:3e:fe:75:a7:9c:d7:45:c9:10:57:b4:
+                    3d:4e:f5:1d:3a:18:23:f1:d4:75:59:fb:cb:9b:d9:
+                    20:5f:03:06:f6:2d:bf:d8:79:6d:45:a7:d3:7b:8e:
+                    fd:55:5a:2c:7d:c8:60:2f:8f:9d:47:7e:9c:33:27:
+                    8e:1c:20:cc:fc:00:7b:ce:e3:d5:11:4a:65:71:0c:
+                    94:de:cb:cf:4a:02:e8:07:4d:0a:9d:6e:72:40:4e:
+                    7c:3c:de:0c:8f:e8:10:41:01:2c:16:05:8e:9d:76:
+                    46:73:d5:cf:db:6e:9f:9c:49:69:29:51:c5:72:74:
+                    83:d8:df:7f:2a:c7:6c:a5:16:ed:1b:ba:6e:a1:b2:
+                    ac:1b:53:1c:48:d1:62:8f:0d:96:b8:9d:79:d6:76:
+                    6f:90:87:fe:c5:4b:3d:39:96:1a:36:d4:db:9d:56:
+                    90:3a:bf:58:a3:a6:05:45:43:9c:af:23:3e:31:65:
+                    22:11:a5:d8:90:63:42:3c:3b:98:f9:76:f0:e7:97:
+                    6a:58:01:0c:fb:56:e9:c7:2f:f6:b3:a6:aa:1a:48:
+                    c1:6f:b5
                 Exponent: 65537 (0x10001)
         X509v3 extensions:
             X509v3 Basic Constraints: 
@@ -28,47 +54,75 @@ Certificate:
             Netscape Comment: 
                 OpenSSL Generated Certificate
             X509v3 Subject Key Identifier: 
-                CB:22:3B:E6:DA:B8:3D:7E:39:61:18:38:50:C8:4D:B4:C8:9A:3E:2B
+                BA:A0:4E:1C:90:A6:1E:61:1B:40:4D:15:4F:2C:54:3E:4E:D4:A3:DC
             X509v3 Authority Key Identifier: 
-                keyid:1C:C7:2B:AA:1B:B1:BB:2E:9A:F4:0F:B1:86:60:57:38:C2:41:05:12
+                keyid:DA:2C:A1:63:9E:33:1D:44:F2:E1:37:17:C9:8F:90:4F:69:F8:B0:85
 
     Signature Algorithm: sha256WithRSAEncryption
-         8e:b7:3d:cc:2b:e5:27:49:49:5a:d4:3c:83:9b:2f:7d:11:de:
-         6b:0f:b4:51:02:e4:37:d0:c4:b5:7b:4b:e3:42:93:75:32:d1:
-         eb:41:a2:27:fe:4d:91:ae:2b:a0:8b:3b:7f:e9:1b:47:85:73:
-         9f:b7:05:74:34:eb:62:12:d8:f2:24:6a:b4:24:58:7a:6a:55:
-         3e:ba:54:f9:51:de:54:0e:19:06:f6:01:23:32:0a:6c:81:e2:
-         57:8a:9e:71:c5:ba:ca:c3:0f:18:d4:ef:54:4b:e1:ee:7c:3a:
-         f3:4d:c3:a3:44:21:9f:c8:ef:85:01:ab:fe:a9:7b:36:05:10:
-         25:5e:61:b8:1d:7c:40:8a:5d:5b:c1:bb:7c:79:45:6e:84:63:
-         b5:3f:51:e9:9c:57:01:de:2d:1e:85:cb:83:f0:16:6d:78:bb:
-         12:01:a5:e5:a2:a7:80:fa:54:8f:9a:5c:de:1c:52:a2:bd:00:
-         49:e4:04:65:30:f9:b9:fc:4f:94:e2:d8:39:89:b6:a5:a5:2d:
-         db:25:a3:0d:5b:f9:e1:2a:13:19:5e:d7:1a:33:89:5b:ac:bd:
-         29:26:da:1a:90:7e:08:19:dd:59:4b:50:fd:46:c3:0b:91:33:
-         8b:c6:70:d1:22:18:37:a3:8b:d3:8f:f8:68:cf:38:8e:e1:ef:
-         ac:17:88:ca
+         5e:39:57:1d:b1:b6:c8:9f:0b:41:b2:a9:42:86:74:c2:f3:41:
+         f0:17:94:93:58:c0:9a:bb:19:48:ef:80:74:c3:12:bb:92:6c:
+         d9:cd:b7:9e:54:ad:5f:ef:38:d5:38:d4:03:49:63:0f:f4:e8:
+         a2:a5:32:e2:f2:53:06:6d:d2:01:48:b9:30:d6:da:1e:b2:b7:
+         44:dc:f2:6d:bd:da:2c:73:b2:4b:d0:c8:df:7e:cf:5c:6a:4c:
+         c6:28:ff:a0:88:43:6d:f2:10:7c:52:06:d6:9a:53:5c:3a:4b:
+         7b:3c:86:1a:11:e1:77:72:c0:d0:fe:ea:67:a1:ab:59:31:57:
+         f3:ea:95:66:34:c1:18:5f:57:5c:63:d0:bd:aa:42:0c:49:84:
+         74:f5:55:b4:d6:b7:48:57:b5:5a:65:fd:14:8a:cf:de:e1:a9:
+         94:68:73:cf:33:96:4b:cb:cb:61:5e:a7:55:58:25:dd:99:39:
+         3f:92:07:11:cf:29:f9:74:fd:de:9d:0b:4f:c0:d1:10:56:2f:
+         63:d6:d7:7d:c1:e5:a2:09:af:bf:1d:ef:37:a8:e6:9b:8f:46:
+         a2:c3:71:f9:f1:8c:1c:b6:3e:8e:b9:22:bc:41:8f:00:ed:f1:
+         10:c8:a2:b9:a9:23:db:59:ec:8c:93:1f:61:50:d8:ce:22:b5:
+         3b:8f:53:3b:73:21:d9:f6:00:bf:9a:2a:96:1b:12:34:d4:c4:
+         93:75:d7:f0:e4:35:49:28:90:02:88:96:b1:4e:04:ac:04:e5:
+         62:cd:a1:5f:63:98:ea:f3:33:95:c7:e6:ac:55:84:65:a6:ae:
+         39:7a:12:66:7e:ff:d2:e7:83:a8:ff:d8:2e:e6:a2:7a:50:da:
+         15:c6:c3:c1:17:47:3d:dd:bc:e0:6e:af:f6:87:5e:c3:47:96:
+         90:31:6c:15:44:05:f9:31:42:1c:8a:40:dc:34:9d:97:01:07:
+         82:bc:cc:5f:cb:e8:f2:b1:55:3b:d0:6d:fb:8c:e2:ee:e9:e5:
+         a1:a2:0f:e1:70:eb:02:70:7a:52:43:47:08:df:46:27:47:7d:
+         47:19:38:53:b7:50:b6:f4:4e:48:d8:05:1c:16:48:27:95:40:
+         c3:b3:9a:05:b1:80:25:9e:31:af:12:a9:12:e9:ff:18:a4:41:
+         63:fb:70:b8:2e:ea:ac:b3:f1:72:17:61:b5:63:72:68:7d:c0:
+         37:6f:4f:55:a1:80:ad:a5:09:3f:6a:f1:07:14:74:72:b4:d4:
+         5b:aa:40:1b:01:2c:53:e1:60:12:bf:f6:fa:4c:44:49:bb:f4:
+         b7:7b:42:48:e2:c8:6d:70:49:03:4c:16:9f:54:72:c7:68:75:
+         c5:39:cd:81:48:8f:0c:b1
 -----BEGIN CERTIFICATE-----
-MIIEETCCAvmgAwIBAgIBATANBgkqhkiG9w0BAQsFADBWMQ8wDQYDVQQDDAZjYWNl
+MIIGlTCCBH2gAwIBAgIBATANBgkqhkiG9w0BAQsFADBWMQ8wDQYDVQQDDAZjYWNl
 cnQxCzAJBgNVBAYTAkZJMREwDwYDVQQIDAhIZWxzaW5raTERMA8GA1UEBwwISGVs
-c2lua2kxEDAOBgNVBAoMB01hcmlhREIwHhcNMTcwNDI1MjA1MjIxWhcNMzcwNDIw
-MjA1MjIxWjCCAUcxCzAJBgNVBAYTAkZJMWEwXwYDVQQIDFhzdGF0ZSBvciBwcm92
+c2lua2kxEDAOBgNVBAoMB01hcmlhREIwHhcNMTgxMjE2MTc1NzAyWhcNMzgxMjEx
+MTc1NzAyWjCCAUcxCzAJBgNVBAYTAkZJMWEwXwYDVQQIDFhzdGF0ZSBvciBwcm92
 aW5jZSB3aXRoaW4gY291bnRyeSwgaW4gb3RoZXIgY2VydGlmaWNhdGVzIGluIHRo
 aXMgZmlsZSBpdCBpcyB0aGUgc2FtZSBhcyBMMUAwPgYDVQQHDDdsb2NhdGlvbiwg
 dXN1YWxseSBhbiBhZGRyZXNzIGJ1dCBvZnRlbiBhbWJpZ3VvdXNseSB1c2VkMTQw
 MgYDVQQKDCtvcmdhbml6YXRpb24gbmFtZSwgdHlwaWNhbGx5IGEgY29tcGFueSBu
 YW1lMUkwRwYDVQQLDEBvcmdhbml6YXRpb25hbCB1bml0IG5hbWUsIGEgZGl2aXNp
 b24gbmFtZSB3aXRoaW4gYW4gb3JnYW5pemF0aW9uMRIwEAYDVQQDDAlsb2NhbGhv
-c3QwgZ8wDQYJKoZIhvcNAQEBBQADgY0AMIGJAoGBAMn+gz8LOIkdQxWTXCa5gGVB
-vS9jZl7b+jMg2cfhNfMUPMi09wnY9bVEj2t+oKQ7RV/m9A8IZ/haTEnk5TkxaYzP
-JXiTpn9Y6ZCcYcsuFLFXshWc6o1vliD+Ke0scbhPH+AFbASxfuC8Qo6/lV6pXsnJ
-pGTCH1mUFMIGRHm80mUtAgMBAAGjezB5MAkGA1UdEwQCMAAwLAYJYIZIAYb4QgEN
-BB8WHU9wZW5TU0wgR2VuZXJhdGVkIENlcnRpZmljYXRlMB0GA1UdDgQWBBTLIjvm
-2rg9fjlhGDhQyE20yJo+KzAfBgNVHSMEGDAWgBQcxyuqG7G7Lpr0D7GGYFc4wkEF
-EjANBgkqhkiG9w0BAQsFAAOCAQEAjrc9zCvlJ0lJWtQ8g5svfRHeaw+0UQLkN9DE
-tXtL40KTdTLR60GiJ/5Nka4roIs7f+kbR4Vzn7cFdDTrYhLY8iRqtCRYempVPrpU
-+VHeVA4ZBvYBIzIKbIHiV4qeccW6ysMPGNTvVEvh7nw6803Do0Qhn8jvhQGr/ql7
-NgUQJV5huB18QIpdW8G7fHlFboRjtT9R6ZxXAd4tHoXLg/AWbXi7EgGl5aKngPpU
-j5pc3hxSor0ASeQEZTD5ufxPlOLYOYm2paUt2yWjDVv54SoTGV7XGjOJW6y9KSba
-GpB+CBndWUtQ/UbDC5Ezi8Zw0SIYN6OL04/4aM84juHvrBeIyg==
+c3QwggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQC1ml/1Tqn9SL608R57
+Bn86MNtuzXiYDg0/k2Fqps09yw/grODAcF7JbrSs11A/jHxqvfUAvgSJNW8hCFEz
+0a+mPTxXi90EDFVyNE5Ii4+z8dHhAS9vqDd8hMhngvbXiw4Qp6VWqv6cL+tQvnU2
++A8OhfrCirYZfNY4KptXyF0jnQ0/uRVa/T1dl8f28PmZXJTxH5PSUyvXPHkHdg1M
+GchchV/y6OsZ2Q/xdRDSgP589NBn+YwwHUbLtpDMleOvAnx1OluJlJkO7sHjJdUm
+5nKzUaT2oaK50WYTzJTNKSfKQ646N52oqaDB01s+XhdOQ7XGYVpVL53i4S5QYg2B
+PMW/VfYOg05uuQBAbt3sddeW6yTRm/D/QrXYfM3U+0h6oQzjsIi8oUyKVpfYh0ca
+R/4+/nWnnNdFyRBXtD1O9R06GCPx1HVZ+8ub2SBfAwb2Lb/YeW1Fp9N7jv1VWix9
+yGAvj51HfpwzJ44cIMz8AHvO49URSmVxDJTey89KAugHTQqdbnJATnw83gyP6BBB
+ASwWBY6ddkZz1c/bbp+cSWkpUcVydIPY338qx2ylFu0bum6hsqwbUxxI0WKPDZa4
+nXnWdm+Qh/7FSz05lho21NudVpA6v1ijpgVFQ5yvIz4xZSIRpdiQY0I8O5j5dvDn
+l2pYAQz7VunHL/azpqoaSMFvtQIDAQABo3sweTAJBgNVHRMEAjAAMCwGCWCGSAGG
++EIBDQQfFh1PcGVuU1NMIEdlbmVyYXRlZCBDZXJ0aWZpY2F0ZTAdBgNVHQ4EFgQU
+uqBOHJCmHmEbQE0VTyxUPk7Uo9wwHwYDVR0jBBgwFoAU2iyhY54zHUTy4TcXyY+Q
+T2n4sIUwDQYJKoZIhvcNAQELBQADggIBAF45Vx2xtsifC0GyqUKGdMLzQfAXlJNY
+wJq7GUjvgHTDEruSbNnNt55UrV/vONU41ANJYw/06KKlMuLyUwZt0gFIuTDW2h6y
+t0Tc8m292ixzskvQyN9+z1xqTMYo/6CIQ23yEHxSBtaaU1w6S3s8hhoR4XdywND+
+6mehq1kxV/PqlWY0wRhfV1xj0L2qQgxJhHT1VbTWt0hXtVpl/RSKz97hqZRoc88z
+lkvLy2Fep1VYJd2ZOT+SBxHPKfl0/d6dC0/A0RBWL2PW133B5aIJr78d7zeo5puP
+RqLDcfnxjBy2Po65IrxBjwDt8RDIormpI9tZ7IyTH2FQ2M4itTuPUztzIdn2AL+a
+KpYbEjTUxJN11/DkNUkokAKIlrFOBKwE5WLNoV9jmOrzM5XH5qxVhGWmrjl6EmZ+
+/9Lng6j/2C7monpQ2hXGw8EXRz3dvOBur/aHXsNHlpAxbBVEBfkxQhyKQNw0nZcB
+B4K8zF/L6PKxVTvQbfuM4u7p5aGiD+Fw6wJwelJDRwjfRidHfUcZOFO3ULb0TkjY
+BRwWSCeVQMOzmgWxgCWeMa8SqRLp/xikQWP7cLgu6qyz8XIXYbVjcmh9wDdvT1Wh
+gK2lCT9q8QcUdHK01FuqQBsBLFPhYBK/9vpMREm79Ld7QkjiyG1wSQNMFp9Ucsdo
+dcU5zYFIjwyx
 -----END CERTIFICATE-----
--- a/mysql-test/std_data/server-key.pem
+++ b/mysql-test/std_data/server-key.pem
@@ -1,15 +1,51 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIICXQIBAAKBgQDJ/oM/CziJHUMVk1wmuYBlQb0vY2Ze2/ozINnH4TXzFDzItPcJ
-2PW1RI9rfqCkO0Vf5vQPCGf4WkxJ5OU5MWmMzyV4k6Z/WOmQnGHLLhSxV7IVnOqN
-b5Yg/intLHG4Tx/gBWwEsX7gvEKOv5VeqV7JyaRkwh9ZlBTCBkR5vNJlLQIDAQAB
-AoGAZzoeMJW3arr9kYUhTtj1+MlDuXf+1PNuRbrAERnSzErHVXrF1M5/owdKlBC+
-X6+6oGwSxavyFkVvP0QusK/D1DxSs60Mk1p3Ax5nVpNYiBdeBULcYiXCSUbKItYL
-vzyECIc02t+auWf/wp3Wy3g6sU6FoT2QCPSsy0qRC48VBzECQQD79P9w0XjJGk8y
-3zPc4JOr3a1UUu4VLjFdFDR2eZZRE02NcXfLHE0+Idk3TYnNDRLoWDJjdfZ0thZZ
-KuJ58wIDAkEAzTxGYmT/aieDC3G8sHMqLUvjN30TfTocv7mWlpxWo7zbRbQm7jsB
-S5weRKtu3yVmdb6rW+5IZSCazc/j4T5tDwJAUgDRSpTCrSFE+Zevt4nYRi2mBjXf
-i0E3i8XRtWWpSMXxjcGKba7ObRRzMA6qdPR2XOynqbtw9Vd2Ops8jthKpQJBAJJm
-8tZxsXlqIiLhyXYdUPLq9XS5tlNYRvXFT9S0RWGb8NbyQesjEDN9dGIL4JUxurs3
-fkmf2ui4r3UtXSNqQqUCQQDjatAhBdibh5UawO5VpI87OJMzATCkY/mX3R+TnbOm
-MkdydjF022P0M4N24DiM+2nBTDp+F45LwtQLa4vmB2No
+MIIJKAIBAAKCAgEAtZpf9U6p/Ui+tPEeewZ/OjDbbs14mA4NP5NhaqbNPcsP4Kzg
+wHBeyW60rNdQP4x8ar31AL4EiTVvIQhRM9Gvpj08V4vdBAxVcjROSIuPs/HR4QEv
+b6g3fITIZ4L214sOEKelVqr+nC/rUL51NvgPDoX6woq2GXzWOCqbV8hdI50NP7kV
+Wv09XZfH9vD5mVyU8R+T0lMr1zx5B3YNTBnIXIVf8ujrGdkP8XUQ0oD+fPTQZ/mM
+MB1Gy7aQzJXjrwJ8dTpbiZSZDu7B4yXVJuZys1Gk9qGiudFmE8yUzSknykOuOjed
+qKmgwdNbPl4XTkO1xmFaVS+d4uEuUGINgTzFv1X2DoNObrkAQG7d7HXXlusk0Zvw
+/0K12HzN1PtIeqEM47CIvKFMilaX2IdHGkf+Pv51p5zXRckQV7Q9TvUdOhgj8dR1
+WfvLm9kgXwMG9i2/2HltRafTe479VVosfchgL4+dR36cMyeOHCDM/AB7zuPVEUpl
+cQyU3svPSgLoB00KnW5yQE58PN4Mj+gQQQEsFgWOnXZGc9XP226fnElpKVHFcnSD
+2N9/KsdspRbtG7puobKsG1McSNFijw2WuJ151nZvkIf+xUs9OZYaNtTbnVaQOr9Y
+o6YFRUOcryM+MWUiEaXYkGNCPDuY+Xbw55dqWAEM+1bpxy/2s6aqGkjBb7UCAwEA
+AQKCAgAi48DR0UYpakNxYe24NxxUYrSVyhYEKJb1AuyxX9CIsa2gPbaEYlJjoshF
+eKxrOSNiIpKijN80OTbbgsyPBxcW+ibQrwdMUs5Lva7Yky+No8Tt1Jt3eTsHyj7y
+XB/uc0HFg1/pkbqO6UntXloIsIm8AZOWsq2fofNKIS1tJQp34CF+f6GRtJuIkrOR
+RuS9SSM4XBe7Tc1bcbvS6iRN3Zo35Ev10Dfy5FgCzLnpp2y9Nw+fGjuLtuQ2GSIi
+nB19Lx4YK5kQiqMJUmefq1ITQi0BNItb8tvflVJmNWTVMbX6Txx3BN5eXBCNjHy4
+M72IhMu0jFmdiiNsRlTujpgHn1RiRkuHStWXnYxh6bM8bnVJ8VpItZTkMH0wqSOn
+CkckFRK6ytr0DwUMovmcw99o6J6HxZI+k9/cjjtnQTHyzj5RN3sSZonMDwMjhAOD
+kKr7MoIOHuINEuiN6giFi3rw7FlBqmk+NolzCG/89J612RmoYYtu2J33hKujwyR2
+tUhvXTApsnA8HUHOQdorXElcNUwS94R26oQ+ZiAjvR/lKmIfApSUAiJiynrV3iYY
+K85FmLXLE4RGDzz69oqnpco4U9WMUl2L1QWiFEgNYUgFFXu5awmRp31CKDXhR8l3
+dm0wXtuFeYE7XNYfSZYFgWgeV8qAtIJoQhA7sonRW4rbQa0d1QKCAQEA33NevILZ
+aFf47WM3GQt7B5OBEPj34tqjd+wqmPXmKAQ43uCLVT5VFz8AEnmW85mqF9zn+lp/
+yieba0Q4S6JU+Z67aC9k/pCWgg9ReAaxBLlcJ3qAzsbrlEDHGBu5iQbVSjJPJHOh
+y3kBxzy12IC0So/79kwyIiQ50l8O83X3Xd2BIWb1valGhbm7FQDWfAL/iTp2jA9h
+qDK84ngiIgH4Mg2opOhyN2hSRAqrkDCfhFIYdhhAAVZCyjQhWo6yJpNYs96QJ22Y
+YvYIxliBnQBwDnwPz3CdTKYckTN5n7ISq73QFvXWTJqdswpSjPidc3Sm3LCnEwwl
+m7d7zHVwmiRvvwKCAQEA0A56NvgZ5aZwp/HFFdgoXBdNFtIDPUOMPJB/I60R/4Nv
+O/+Z/K1frgeApyQm65KVwK1ct5p4PoyPiuKZ8RY41IGOJJ6NN/cz9U9RaE23qPgo
+vJDUpYkPCtnXotsvATst/q9LETGf0Wr6LBupHaRWaUpHxWmdhyKpmBrHRlv9J7qk
+0WgdqaarNHg3sIrNA+eDBq3ZEGppNj5voEUdhAefomEutECj+wfvcrDgaxiZ6rE1
+N/SA1On8mo4rAFtIRynT3iOm7N5ft/ypFro9FiuCs3j63wMicDlfRlTPnLSA4tSV
+YunduqSGuwkHQi17sPxKLj1eR3mP3+RLsGYYDbf9iwKCAQAHH11QsvcNHWjA4F28
+PaJxTmFZ0iXMpF17kBqXf5cThXbum8e1MuRxdj+hVyDZqPdEBXWbhFD4d6S7mZQV
+zRnaTxX4LzKP7rntcWGOauAvn1zxoTUD4lFWC26NNS0H0iDwWYQuTEVHABdva178
+hLW6frZwVmNutSrmqsMGPadh5ZKdJvQEVKX70zzEYZ90mn/ctvFaa48YVm2w5lVo
+MhngNpE+mG4zl+6vXouzb+lVxWwkL4eywzYjzd+WTGAfXKkCKsD7kMG/vcnv0Jhp
+Eb+AprG2UgehjtXwRuzEi1+O++DrKnDnRjJqp5SUC8SxDnHXLQZWVBfKy+F1VfT9
+5NVvAoIBAQCZwek+EHKLDkr5QqqFLz78d9fnwoLUVsQfyiW7ht7jNw4QiCAT5Ujt
+guvoGcaNl8IGJ9h6c7TjfNtkJ3ze964+FhSnKWlx78W9R8YA0Fanj+YDHQGZ6EVu
+nnlHCLafMU4XhxJx/Xt/t5B04mhD2LxniZaq5Ree09YD695TOA5JuxytCUXKp7qk
+xydY1+sVNJXXbMka4ZkpllDGX+2lMkEeLrXmTlMhZMbnuqHPY9nz4MxUsuQ29UlS
+KzjKRauPs/C4nR5ytRCGFzoa6r3UUYmA+ypjf5yG1U982Zao15jciYBxHiZYFdjg
+BpcPqC9eJ8TmkUHIjqTuMHhjP8AM+2lrAoIBAAjEh3ubLuMlCgcud0jn7yqvZkBr
+nI5H2tpNRqbwvR5O/nkdRZ3z5AcwH5eJJTmWLXK1OWAOm5IaI6LodiDTPSdnX/Aw
+mNdnIhSSrTFY9WHRRwJHp8V7GqBPO+R6sq4/2PnAWQY6CsAG+yjc1qb9BO1PgnYJ
+TwuHj6gfUzWz/P1lAMc4fm3ODtNsEkmiitF8DKII2HId96WilaXdAKQrja7Su1Iy
+jBdpDIBDQI9WuO/UhZxlilWwUjy5vBtE8p0Oc4rwSMZBVx14lhdbxEBFgYLLxV2J
+ELsNuuImXhPVDnre/f63wpo0IvAH5dRd+eEC40RRbLrxoo6DQiD3DS9RCfE=
 -----END RSA PRIVATE KEY-----
--- a/mysql-test/std_data/server8k-cert.pem
+++ b/mysql-test/std_data/server8k-cert.pem
@@ -5,82 +5,82 @@ Certificate:
     Signature Algorithm: sha256WithRSAEncryption
         Issuer: CN=cacert, C=FI, ST=Helsinki, L=Helsinki, O=MariaDB
         Validity
-            Not Before: Apr 25 20:52:33 2017 GMT
-            Not After : Apr 20 20:52:33 2037 GMT
+            Not Before: Dec 16 17:57:11 2018 GMT
+            Not After : Dec 11 17:57:11 2038 GMT
         Subject: C=FI, ST=Helsinki, L=Helsinki, O=MariaDB, CN=server8k
         Subject Public Key Info:
             Public Key Algorithm: rsaEncryption
                 Public-Key: (8192 bit)
                 Modulus:
-                    00:b3:a7:81:cf:a3:9b:3c:57:2a:8e:de:13:08:a6:
-                    15:6a:68:08:ce:a1:a0:ef:fc:32:95:5a:9a:b5:cc:
-                    84:bb:a3:15:a0:fc:29:bb:71:0e:f3:40:20:97:35:
-                    92:92:89:36:6a:5f:be:a5:24:5a:c8:0e:f7:53:a8:
-                    e0:74:05:90:38:21:a5:25:72:2b:56:70:74:aa:d6:
-                    30:25:e0:95:dd:3d:4a:48:4b:25:a7:a8:c4:eb:e5:
-                    d6:10:4f:95:42:91:b0:cb:68:2f:bf:96:0f:0a:9b:
-                    2b:01:0d:03:ae:3f:b7:7e:f6:1e:c1:14:42:04:7f:
-                    09:21:e8:3e:87:c0:b3:d7:dd:4f:7e:95:b6:83:33:
-                    8f:cc:f4:bb:11:8e:66:5f:b4:32:22:8b:8e:34:93:
-                    82:0f:02:d1:6f:85:b2:a7:7e:36:0c:f3:0d:91:46:
-                    f9:a5:91:dc:60:4a:0d:eb:e5:37:e3:76:d8:13:bb:
-                    55:33:23:4c:b3:90:df:6a:b6:80:3d:f6:bd:9d:bd:
-                    ef:d3:8d:7a:9a:61:ac:02:3d:10:b5:c2:53:d2:7e:
-                    ee:1c:90:a5:b0:e8:db:71:8f:d3:53:e4:80:b0:b6:
-                    cc:b0:f2:eb:46:c3:d9:3c:48:01:1d:3a:5c:d0:f2:
-                    17:9c:9a:8e:fc:2a:36:e7:1c:59:41:66:ad:e1:d3:
-                    82:d8:95:ee:ae:dc:ab:72:94:cb:0b:8a:df:ef:70:
-                    f7:f9:e2:77:76:64:67:2d:02:dc:1a:db:02:e9:5e:
-                    63:3f:3e:07:e9:65:e4:02:78:62:55:e1:52:18:5a:
-                    a3:9c:56:dd:cd:d3:f7:f2:55:45:62:20:ce:fd:41:
-                    93:b9:c2:20:90:23:a4:41:4f:30:5c:5e:a0:97:6f:
-                    a4:1f:87:6a:97:87:70:f9:d5:56:b7:6a:cc:f1:28:
-                    d2:26:34:57:65:9c:2d:41:43:8b:6c:bd:15:6f:ec:
-                    ce:3e:9f:4d:2f:a6:21:d8:f7:4d:d1:63:4f:f0:9e:
-                    04:c3:82:33:58:34:20:4b:81:0d:42:0d:44:62:48:
-                    1b:4c:cf:e0:69:e4:c8:f8:9c:98:1f:74:37:3a:3f:
-                    23:40:3a:88:80:c2:9c:9b:ba:6a:b9:2c:3a:67:a6:
-                    b6:5c:83:fd:93:3f:6e:53:19:48:e3:3d:fa:f5:8e:
-                    03:40:8f:e6:af:c5:de:aa:f8:70:77:36:2b:6e:3a:
-                    df:bc:d7:f6:c7:4f:02:7f:a1:ad:2f:3c:e9:08:45:
-                    2f:57:1a:f1:5b:9c:27:07:3a:44:43:3e:05:b1:4a:
-                    1b:42:a4:90:ab:ce:4b:f7:c1:9b:d7:dd:6f:05:11:
-                    f9:e8:a3:a1:99:6c:2e:27:3d:b1:54:6f:eb:b2:e8:
-                    f8:e6:12:6f:8a:69:05:cc:c6:f6:c4:2d:b0:06:b6:
-                    21:50:28:50:dc:8c:b2:8c:65:ce:d9:2b:51:b8:62:
-                    56:70:b4:46:4d:e4:4c:a1:86:31:3d:e3:07:7f:73:
-                    e3:09:6e:ee:a7:29:2a:80:23:8a:b6:0c:22:9e:43:
-                    1c:a9:8c:47:2c:02:6d:13:ef:88:a7:d7:d1:60:d9:
-                    6b:28:80:97:8a:b7:ff:62:6b:28:54:0d:ab:63:3a:
-                    ac:57:9e:2f:95:2b:f4:a0:3d:a1:60:a0:57:1a:09:
-                    23:ee:fc:4d:9b:45:83:f2:0a:1f:b3:77:0e:98:20:
-                    b3:16:fc:97:03:30:fc:ab:3e:7e:fc:3e:76:64:cb:
-                    4b:4d:c9:6c:8f:aa:70:ec:03:c5:81:31:43:11:9a:
-                    fc:2d:ba:25:94:0b:73:7e:c1:39:01:ca:68:70:1a:
-                    ef:a4:ba:dc:13:d6:37:17:09:0d:ca:aa:21:94:bb:
-                    3e:43:fd:8c:80:e6:58:fd:75:23:5a:ca:1f:0b:2e:
-                    35:c2:cd:77:0c:73:b3:2f:9a:e4:03:68:58:c8:3f:
-                    f3:76:8d:05:53:d2:34:db:09:af:70:ed:67:0b:ff:
-                    2f:64:35:2d:71:22:7a:f2:f2:2b:7d:6d:be:80:bd:
-                    18:38:37:01:67:a4:5f:48:29:52:78:9f:b5:b8:80:
-                    70:ca:10:6a:e1:27:3b:c0:61:4e:96:f8:e8:0f:42:
-                    b1:fa:bc:05:ef:31:91:38:ab:0f:a7:ef:2c:86:c1:
-                    ab:85:36:7c:10:9d:c7:0a:6b:e6:2b:11:dd:fa:45:
-                    e7:d2:c3:3a:c1:cd:60:27:68:26:d7:87:49:5f:e3:
-                    7f:26:20:1f:99:a8:f9:00:a6:0d:d5:13:81:6e:fc:
-                    27:8a:03:58:6b:0e:3a:c7:20:a1:5c:24:d1:08:a6:
-                    5a:1e:e7:81:3d:21:05:2e:4b:5d:18:14:f1:4f:ab:
-                    5c:e5:81:da:c1:2c:f6:12:ff:25:09:97:1e:9d:51:
-                    6d:1f:63:2e:9c:ed:7d:86:45:b0:35:73:f6:9f:82:
-                    65:bc:7f:46:88:e1:a1:60:a1:37:50:26:9c:d7:76:
-                    f3:dc:94:89:6d:d7:be:a6:17:a1:2b:71:14:37:17:
-                    5f:ed:5a:39:99:fc:69:af:bb:63:55:c6:60:7b:64:
-                    d8:bb:b7:4a:9f:b9:aa:89:4b:30:1f:9c:ef:23:7a:
-                    af:7e:31:dd:fb:dd:0a:d5:04:0a:8c:57:6f:64:80:
-                    eb:f3:ee:6c:33:d2:72:c2:24:68:a2:4e:99:88:1c:
-                    f7:3b:5d:10:cc:4e:ea:a5:cb:00:40:8a:a9:63:2e:
-                    18:92:eb:b1:28:05:fe:19:ea:7b:32:fb:63:56:2c:
-                    0b:20:01:92:a3
+                    00:c6:ae:72:fe:d0:f8:84:c6:1c:ac:a6:c7:7d:fa:
+                    d2:3e:27:ad:ac:2f:39:0a:1d:4f:ca:1f:47:99:dc:
+                    c4:09:6e:d0:e0:52:b1:15:46:4a:68:a4:bb:59:86:
+                    59:47:3c:ae:28:0a:ad:67:b5:42:16:a3:0b:e6:e7:
+                    0d:cd:2a:e3:8b:56:9c:61:14:2a:4f:e5:54:51:20:
+                    e9:e6:cd:c1:ce:8b:2a:8c:17:20:18:7d:1f:29:cc:
+                    44:a5:fd:c0:40:8e:6c:91:7f:08:b6:61:2e:3d:6e:
+                    30:12:75:38:21:ac:ac:74:05:bf:68:f4:2a:b3:c1:
+                    53:41:50:3d:d3:de:25:e3:14:af:1e:19:91:08:74:
+                    7b:20:c8:e1:b4:00:2f:ec:e4:97:c5:03:fc:82:c8:
+                    c5:65:41:26:a9:b4:1b:e5:bc:58:07:a9:18:9f:48:
+                    26:09:45:37:27:a1:30:19:55:19:ec:98:00:bb:8b:
+                    fb:42:eb:56:ab:23:f5:d4:e0:7c:e5:4a:a8:74:8e:
+                    54:aa:c1:ea:0f:3a:a9:f3:1c:cd:63:47:b5:49:22:
+                    69:f3:cf:86:cf:bc:4e:be:24:cd:37:bc:c7:18:1e:
+                    a7:0e:b5:7b:fc:da:48:ef:de:9d:db:a0:d7:9c:1b:
+                    cf:dc:9c:24:3e:6e:ef:42:92:34:93:de:e2:fb:4b:
+                    5e:e5:46:62:89:3a:ec:55:b2:29:34:fb:1d:9f:63:
+                    8a:1b:82:b9:88:64:19:8d:b5:72:21:28:56:8e:f5:
+                    9f:44:1a:c8:6c:05:ec:01:52:88:d8:04:d1:24:e0:
+                    eb:19:22:49:aa:dc:04:3c:db:14:b0:4b:33:f2:38:
+                    22:86:dd:99:e5:a9:43:f3:0d:6f:45:ca:cc:23:f7:
+                    89:42:7e:1b:f1:62:15:68:f8:09:a9:09:65:75:06:
+                    9f:06:f5:28:c4:07:61:25:73:69:b8:75:a1:ba:46:
+                    a0:cf:3a:86:69:92:3b:2e:60:2f:cf:69:f4:9e:e6:
+                    4e:bb:05:5b:6e:f6:50:41:02:fc:e5:ae:33:36:ad:
+                    07:fa:8a:57:c4:d7:0a:28:9f:c5:7b:04:71:8e:d0:
+                    60:52:e2:dc:1d:a7:7c:aa:43:00:07:98:4e:fd:ec:
+                    65:da:85:3c:14:a1:63:1f:c8:e4:36:ea:e0:13:da:
+                    68:e1:91:4e:7d:7f:30:93:3b:1c:22:c9:24:59:1f:
+                    0c:1d:7a:1e:f8:49:3a:e7:27:42:ad:53:7d:5c:d6:
+                    1f:71:5d:d2:04:f6:42:05:c5:bd:69:a0:c5:fd:7a:
+                    d0:c9:15:ae:dc:02:df:d5:66:37:27:2d:91:9d:a2:
+                    d5:62:54:c8:a6:b1:cc:38:1a:37:54:ca:11:67:f5:
+                    36:5d:0b:e6:36:c6:9c:ea:e5:31:e6:c3:92:76:3f:
+                    fc:1c:7d:de:77:d3:d2:e8:8e:cb:5f:e1:b3:16:f0:
+                    6c:66:13:ab:85:de:67:ab:7d:be:fd:33:cc:c6:ae:
+                    68:7e:80:69:b0:db:55:06:f0:4d:5f:34:5d:7b:06:
+                    84:0d:64:f6:b7:ad:4a:8d:2e:af:4a:b6:83:34:0d:
+                    50:a0:9f:8a:d6:44:2e:d0:37:8e:cf:93:11:7b:c3:
+                    95:71:22:27:6f:3f:36:fe:75:cf:ff:bc:c7:87:42:
+                    35:bb:59:02:eb:f1:89:11:46:5c:94:22:d2:fe:40:
+                    f1:a4:98:e2:71:a5:75:d1:03:2e:12:98:54:30:81:
+                    75:81:96:91:8b:74:56:dd:7f:ba:23:22:6e:e1:75:
+                    c8:65:ed:df:46:40:c5:80:14:75:67:b6:7b:ea:1d:
+                    96:34:c6:cb:d7:bb:56:6f:1e:2c:3f:f5:34:03:f1:
+                    00:97:0b:cc:66:06:04:2c:1f:71:06:0c:02:17:f4:
+                    b5:15:75:59:cf:d5:47:14:88:aa:8c:65:64:6d:87:
+                    49:e4:7b:aa:a1:45:23:97:13:e8:45:f4:9a:ac:47:
+                    6b:4f:e0:12:23:5f:6d:bc:16:68:53:e4:c1:19:4b:
+                    f3:7e:64:54:ca:d3:4f:a7:fe:40:33:7e:30:2a:66:
+                    98:3b:c3:09:c6:0d:ce:d4:0c:dc:36:c7:cc:47:56:
+                    d1:48:c5:a8:73:ac:0e:7a:e9:c0:7f:9c:9b:52:71:
+                    be:85:1a:39:0e:02:8c:21:85:a3:8f:a7:14:c7:9b:
+                    5c:d4:91:06:fc:9f:fa:65:86:0a:f4:92:73:84:cb:
+                    7d:24:31:a2:58:79:45:23:8b:19:93:d4:df:5c:67:
+                    ec:7d:e3:20:89:03:09:1e:a0:6d:ed:fa:ef:93:de:
+                    29:6c:1b:0b:ab:0b:b0:ad:13:ae:83:3d:78:e4:d0:
+                    4f:aa:14:60:cf:10:3d:93:8a:af:c8:04:2b:1d:39:
+                    a6:7f:94:f2:f1:1f:f4:72:c4:5c:12:32:4a:1b:d6:
+                    0b:4d:3a:e0:a3:8d:95:2a:44:ad:26:eb:96:cc:aa:
+                    d5:5f:82:c1:55:47:29:e1:c7:6d:64:b9:0f:12:2b:
+                    56:f8:4c:e7:d2:13:4f:64:bc:90:cd:8a:ae:09:ec:
+                    ef:5a:95:5a:df:b0:8b:71:4e:2d:28:26:fb:9e:84:
+                    4f:b2:97:ad:c2:3b:ec:0e:f6:15:01:ed:66:fd:49:
+                    cc:ff:c4:bb:78:e7:b5:6f:d8:6b:c8:95:d2:6e:c2:
+                    56:91:ae:98:ee:4f:1c:e3:b6:ad:43:39:38:83:fa:
+                    04:0a:b0:76:70:17:59:3e:4f:10:90:7d:6a:0a:8c:
+                    e2:7f:23:26:cb
                 Exponent: 65537 (0x10001)
         X509v3 extensions:
             X509v3 Basic Constraints: 
@@ -88,61 +88,80 @@ Certificate:
             Netscape Comment: 
                 OpenSSL Generated Certificate
             X509v3 Subject Key Identifier: 
-                43:3B:26:15:A5:60:62:3F:07:2A:8C:8C:DA:F8:74:BE:36:B2:33:E4
+                7B:42:D0:32:08:49:6C:D2:8A:5E:1C:69:73:5B:D9:64:72:FB:80:63
             X509v3 Authority Key Identifier: 
-                keyid:1C:C7:2B:AA:1B:B1:BB:2E:9A:F4:0F:B1:86:60:57:38:C2:41:05:12
+                keyid:DA:2C:A1:63:9E:33:1D:44:F2:E1:37:17:C9:8F:90:4F:69:F8:B0:85
 
     Signature Algorithm: sha256WithRSAEncryption
-         5c:c7:4e:56:eb:21:1b:f3:5d:c4:56:fa:bf:6d:e1:0d:fa:cc:
-         b0:7c:13:58:b9:d4:47:b3:de:b4:ae:f2:45:02:88:65:43:89:
-         cf:ea:9c:d0:ea:45:42:35:d2:ec:bc:b4:73:03:db:85:7b:c7:
-         f0:68:d7:dc:2b:70:71:63:ed:33:68:9f:29:ff:67:0a:69:3b:
-         c4:65:7c:25:00:cf:7f:ee:aa:fe:44:19:70:f2:74:db:da:9f:
-         52:92:1a:03:e6:0a:49:85:2e:2a:02:c1:81:d4:6d:cf:98:d1:
-         e9:6a:2f:bb:fa:a6:d5:4b:55:38:c2:aa:8c:f5:d3:f9:e6:74:
-         db:00:d6:29:f9:d1:9c:7a:0e:98:c5:e5:8c:16:49:d8:cf:ee:
-         a0:cb:c2:2c:31:cf:62:2c:b1:7f:6d:60:b0:ce:d5:07:45:e8:
-         44:17:7c:c4:12:fb:00:57:24:57:e8:17:78:3b:1e:0d:91:aa:
-         67:98:d0:e5:9a:86:4b:88:4a:2f:55:5d:2e:13:ec:f5:4a:7d:
-         e7:13:a7:57:03:41:2f:f6:6c:08:8e:58:ef:b0:7a:79:32:e3:
-         bb:4b:eb:4d:42:cd:42:96:2d:67:f6:4c:c1:f6:62:fb:c0:3e:
-         09:69:8f:36:7e:fa:c8:cd:ff:72:3e:df:92:f7:8f:44:cf:77:
-         fe:6e:74:de
+         6e:bc:22:31:54:f0:8d:b4:a7:6a:5f:8e:86:59:0e:69:59:0c:
+         e0:3f:d1:8e:76:2e:81:f1:a1:bb:b5:74:c8:6e:6e:04:d4:40:
+         9a:7f:07:f3:79:fd:f6:5b:ff:62:86:b9:84:09:07:e2:4a:11:
+         90:a5:fc:38:c1:22:19:a8:ac:d3:f8:61:3a:0e:a1:5a:9b:78:
+         fa:a2:d7:0b:5e:b0:9c:0b:21:88:26:a3:19:31:9a:e3:00:0a:
+         f1:bf:e2:14:6b:c4:b9:f9:e1:9e:d0:12:ce:c7:bc:0b:04:d9:
+         0d:48:80:0e:c5:b3:49:87:5f:49:34:08:10:2d:66:6e:ea:d8:
+         60:6e:6f:47:f6:fe:45:99:92:54:01:2a:0f:36:22:6e:89:9b:
+         ae:ea:75:6f:87:27:f1:16:7f:f5:b1:bf:20:31:02:73:af:d4:
+         f1:29:e7:7a:1f:6a:24:3f:51:17:03:88:d9:e6:ab:3a:3d:42:
+         d8:bc:db:57:6f:7f:8a:a6:14:ac:65:54:ec:37:ad:1d:ab:25:
+         9e:b1:b1:f7:b4:24:34:fb:a2:c6:33:c4:db:fa:f2:e4:c5:6f:
+         19:c5:3d:a0:72:a8:fb:a3:4c:5d:38:90:04:87:f7:f5:e9:f9:
+         93:8f:f1:c5:57:75:37:7c:e1:94:e9:90:2b:b9:49:83:c0:f9:
+         56:d9:8a:61:1b:ef:b8:ff:1e:7b:39:6a:dd:70:bb:ef:cf:df:
+         10:2a:fd:56:b3:a0:e8:ea:dd:79:51:fb:e0:07:d5:69:ea:e5:
+         29:12:e7:51:b1:55:d4:ee:33:cb:21:b9:62:66:4a:f0:78:3b:
+         e1:e1:bd:e4:0e:58:a9:db:fc:5d:1c:c8:ac:fa:56:84:8c:e2:
+         08:5f:3b:08:17:00:6c:fe:16:a1:c3:44:e9:d4:09:48:a5:75:
+         9e:cc:02:07:86:9d:63:17:e5:5a:9b:4c:9a:a2:8b:6f:79:c1:
+         05:52:90:73:cb:df:62:46:8e:40:af:63:52:b1:2d:63:a0:39:
+         6c:53:d1:7d:6a:97:f4:17:7e:45:c7:cc:65:bd:4a:ee:f0:f6:
+         0c:a1:75:9b:cc:5c:d8:9c:da:0a:35:e5:b9:35:b2:9a:25:1b:
+         3b:1a:79:6a:be:5e:e9:2a:d6:1d:81:63:f4:81:8c:98:de:0d:
+         c0:e5:22:98:53:6f:8e:9f:22:5a:de:31:90:75:d7:10:3c:bd:
+         de:b6:40:e8:2a:e9:c6:e2:73:5d:8d:b5:8d:01:9e:cd:d8:d0:
+         e2:e6:46:3b:74:ac:6f:04:f2:6c:c4:33:cc:a5:e8:ff:e3:ec:
+         02:58:5a:53:e9:2e:26:de:2c:00:35:13:20:47:4b:4d:0f:8c:
+         fc:b1:af:e7:a7:94:b7:8f
 -----BEGIN CERTIFICATE-----
-MIIGpDCCBYygAwIBAgIBAjANBgkqhkiG9w0BAQsFADBWMQ8wDQYDVQQDDAZjYWNl
+MIIHpDCCBYygAwIBAgIBAjANBgkqhkiG9w0BAQsFADBWMQ8wDQYDVQQDDAZjYWNl
 cnQxCzAJBgNVBAYTAkZJMREwDwYDVQQIDAhIZWxzaW5raTERMA8GA1UEBwwISGVs
-c2lua2kxEDAOBgNVBAoMB01hcmlhREIwHhcNMTcwNDI1MjA1MjMzWhcNMzcwNDIw
-MjA1MjMzWjBYMQswCQYDVQQGEwJGSTERMA8GA1UECAwISGVsc2lua2kxETAPBgNV
+c2lua2kxEDAOBgNVBAoMB01hcmlhREIwHhcNMTgxMjE2MTc1NzExWhcNMzgxMjEx
+MTc1NzExWjBYMQswCQYDVQQGEwJGSTERMA8GA1UECAwISGVsc2lua2kxETAPBgNV
 BAcMCEhlbHNpbmtpMRAwDgYDVQQKDAdNYXJpYURCMREwDwYDVQQDDAhzZXJ2ZXI4
-azCCBCIwDQYJKoZIhvcNAQEBBQADggQPADCCBAoCggQBALOngc+jmzxXKo7eEwim
-FWpoCM6hoO/8MpVamrXMhLujFaD8KbtxDvNAIJc1kpKJNmpfvqUkWsgO91Oo4HQF
-kDghpSVyK1ZwdKrWMCXgld09SkhLJaeoxOvl1hBPlUKRsMtoL7+WDwqbKwENA64/
-t372HsEUQgR/CSHoPofAs9fdT36VtoMzj8z0uxGOZl+0MiKLjjSTgg8C0W+Fsqd+
-NgzzDZFG+aWR3GBKDevlN+N22BO7VTMjTLOQ32q2gD32vZ2979ONepphrAI9ELXC
-U9J+7hyQpbDo23GP01PkgLC2zLDy60bD2TxIAR06XNDyF5yajvwqNuccWUFmreHT
-gtiV7q7cq3KUywuK3+9w9/nid3ZkZy0C3BrbAuleYz8+B+ll5AJ4YlXhUhhao5xW
-3c3T9/JVRWIgzv1Bk7nCIJAjpEFPMFxeoJdvpB+HapeHcPnVVrdqzPEo0iY0V2Wc
-LUFDi2y9FW/szj6fTS+mIdj3TdFjT/CeBMOCM1g0IEuBDUINRGJIG0zP4GnkyPic
-mB90Nzo/I0A6iIDCnJu6arksOmemtlyD/ZM/blMZSOM9+vWOA0CP5q/F3qr4cHc2
-K24637zX9sdPAn+hrS886QhFL1ca8VucJwc6REM+BbFKG0KkkKvOS/fBm9fdbwUR
-+eijoZlsLic9sVRv67Lo+OYSb4ppBczG9sQtsAa2IVAoUNyMsoxlztkrUbhiVnC0
-Rk3kTKGGMT3jB39z4wlu7qcpKoAjirYMIp5DHKmMRywCbRPviKfX0WDZayiAl4q3
-/2JrKFQNq2M6rFeeL5Ur9KA9oWCgVxoJI+78TZtFg/IKH7N3Dpggsxb8lwMw/Ks+
-fvw+dmTLS03JbI+qcOwDxYExQxGa/C26JZQLc37BOQHKaHAa76S63BPWNxcJDcqq
-IZS7PkP9jIDmWP11I1rKHwsuNcLNdwxzsy+a5ANoWMg/83aNBVPSNNsJr3DtZwv/
-L2Q1LXEievLyK31tvoC9GDg3AWekX0gpUniftbiAcMoQauEnO8BhTpb46A9Csfq8
-Be8xkTirD6fvLIbBq4U2fBCdxwpr5isR3fpF59LDOsHNYCdoJteHSV/jfyYgH5mo
-+QCmDdUTgW78J4oDWGsOOscgoVwk0QimWh7ngT0hBS5LXRgU8U+rXOWB2sEs9hL/
-JQmXHp1RbR9jLpztfYZFsDVz9p+CZbx/RojhoWChN1AmnNd289yUiW3XvqYXoStx
-FDcXX+1aOZn8aa+7Y1XGYHtk2Lu3Sp+5qolLMB+c7yN6r34x3fvdCtUECoxXb2SA
-6/PubDPScsIkaKJOmYgc9ztdEMxO6qXLAECKqWMuGJLrsSgF/hnqezL7Y1YsCyAB
-kqMCAwEAAaN7MHkwCQYDVR0TBAIwADAsBglghkgBhvhCAQ0EHxYdT3BlblNTTCBH
-ZW5lcmF0ZWQgQ2VydGlmaWNhdGUwHQYDVR0OBBYEFEM7JhWlYGI/ByqMjNr4dL42
-sjPkMB8GA1UdIwQYMBaAFBzHK6obsbsumvQPsYZgVzjCQQUSMA0GCSqGSIb3DQEB
-CwUAA4IBAQBcx05W6yEb813EVvq/beEN+sywfBNYudRHs960rvJFAohlQ4nP6pzQ
-6kVCNdLsvLRzA9uFe8fwaNfcK3BxY+0zaJ8p/2cKaTvEZXwlAM9/7qr+RBlw8nTb
-2p9SkhoD5gpJhS4qAsGB1G3PmNHpai+7+qbVS1U4wqqM9dP55nTbANYp+dGceg6Y
-xeWMFknYz+6gy8IsMc9iLLF/bWCwztUHRehEF3zEEvsAVyRX6Bd4Ox4NkapnmNDl
-moZLiEovVV0uE+z1Sn3nE6dXA0Ev9mwIjljvsHp5MuO7S+tNQs1Cli1n9kzB9mL7
-wD4JaY82fvrIzf9yPt+S949Ez3f+bnTe
+azCCBCIwDQYJKoZIhvcNAQEBBQADggQPADCCBAoCggQBAMaucv7Q+ITGHKymx336
+0j4nrawvOQodT8ofR5ncxAlu0OBSsRVGSmiku1mGWUc8rigKrWe1QhajC+bnDc0q
+44tWnGEUKk/lVFEg6ebNwc6LKowXIBh9HynMRKX9wECObJF/CLZhLj1uMBJ1OCGs
+rHQFv2j0KrPBU0FQPdPeJeMUrx4ZkQh0eyDI4bQAL+zkl8UD/ILIxWVBJqm0G+W8
+WAepGJ9IJglFNyehMBlVGeyYALuL+0LrVqsj9dTgfOVKqHSOVKrB6g86qfMczWNH
+tUkiafPPhs+8Tr4kzTe8xxgepw61e/zaSO/endug15wbz9ycJD5u70KSNJPe4vtL
+XuVGYok67FWyKTT7HZ9jihuCuYhkGY21ciEoVo71n0QayGwF7AFSiNgE0STg6xki
+SarcBDzbFLBLM/I4IobdmeWpQ/MNb0XKzCP3iUJ+G/FiFWj4CakJZXUGnwb1KMQH
+YSVzabh1obpGoM86hmmSOy5gL89p9J7mTrsFW272UEEC/OWuMzatB/qKV8TXCiif
+xXsEcY7QYFLi3B2nfKpDAAeYTv3sZdqFPBShYx/I5Dbq4BPaaOGRTn1/MJM7HCLJ
+JFkfDB16HvhJOucnQq1TfVzWH3Fd0gT2QgXFvWmgxf160MkVrtwC39VmNyctkZ2i
+1WJUyKaxzDgaN1TKEWf1Nl0L5jbGnOrlMebDknY//Bx93nfT0uiOy1/hsxbwbGYT
+q4XeZ6t9vv0zzMauaH6AabDbVQbwTV80XXsGhA1k9retSo0ur0q2gzQNUKCfitZE
+LtA3js+TEXvDlXEiJ28/Nv51z/+8x4dCNbtZAuvxiRFGXJQi0v5A8aSY4nGlddED
+LhKYVDCBdYGWkYt0Vt1/uiMibuF1yGXt30ZAxYAUdWe2e+odljTGy9e7Vm8eLD/1
+NAPxAJcLzGYGBCwfcQYMAhf0tRV1Wc/VRxSIqoxlZG2HSeR7qqFFI5cT6EX0mqxH
+a0/gEiNfbbwWaFPkwRlL835kVMrTT6f+QDN+MCpmmDvDCcYNztQM3DbHzEdW0UjF
+qHOsDnrpwH+cm1JxvoUaOQ4CjCGFo4+nFMebXNSRBvyf+mWGCvSSc4TLfSQxolh5
+RSOLGZPU31xn7H3jIIkDCR6gbe3675PeKWwbC6sLsK0TroM9eOTQT6oUYM8QPZOK
+r8gEKx05pn+U8vEf9HLEXBIyShvWC0064KONlSpErSbrlsyq1V+CwVVHKeHHbWS5
+DxIrVvhM59ITT2S8kM2Krgns71qVWt+wi3FOLSgm+56ET7KXrcI77A72FQHtZv1J
+zP/Eu3jntW/Ya8iV0m7CVpGumO5PHOO2rUM5OIP6BAqwdnAXWT5PEJB9agqM4n8j
+JssCAwEAAaN7MHkwCQYDVR0TBAIwADAsBglghkgBhvhCAQ0EHxYdT3BlblNTTCBH
+ZW5lcmF0ZWQgQ2VydGlmaWNhdGUwHQYDVR0OBBYEFHtC0DIISWzSil4caXNb2WRy
++4BjMB8GA1UdIwQYMBaAFNosoWOeMx1E8uE3F8mPkE9p+LCFMA0GCSqGSIb3DQEB
+CwUAA4ICAQBuvCIxVPCNtKdqX46GWQ5pWQzgP9GOdi6B8aG7tXTIbm4E1ECafwfz
+ef32W/9ihrmECQfiShGQpfw4wSIZqKzT+GE6DqFam3j6otcLXrCcCyGIJqMZMZrj
+AArxv+IUa8S5+eGe0BLOx7wLBNkNSIAOxbNJh19JNAgQLWZu6thgbm9H9v5FmZJU
+ASoPNiJuiZuu6nVvhyfxFn/1sb8gMQJzr9TxKed6H2okP1EXA4jZ5qs6PULYvNtX
+b3+KphSsZVTsN60dqyWesbH3tCQ0+6LGM8Tb+vLkxW8ZxT2gcqj7o0xdOJAEh/f1
+6fmTj/HFV3U3fOGU6ZAruUmDwPlW2YphG++4/x57OWrdcLvvz98QKv1Ws6Do6t15
+UfvgB9Vp6uUpEudRsVXU7jPLIbliZkrweDvh4b3kDlip2/xdHMis+laEjOIIXzsI
+FwBs/hahw0Tp1AlIpXWezAIHhp1jF+Vam0yaootvecEFUpBzy99iRo5Ar2NSsS1j
+oDlsU9F9apf0F35Fx8xlvUru8PYMoXWbzFzYnNoKNeW5NbKaJRs7Gnlqvl7pKtYd
+gWP0gYyY3g3A5SKYU2+OnyJa3jGQddcQPL3etkDoKunG4nNdjbWNAZ7N2NDi5kY7
+dKxvBPJsxDPMpej/4+wCWFpT6S4m3iwANRMgR0tND4z8sa/np5S3jw==
 -----END CERTIFICATE-----
--- a/mysql-test/std_data/server8k-key.pem
+++ b/mysql-test/std_data/server8k-key.pem
@@ -1,99 +1,99 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIISKAIBAAKCBAEAs6eBz6ObPFcqjt4TCKYVamgIzqGg7/wylVqatcyEu6MVoPwp
-u3EO80AglzWSkok2al++pSRayA73U6jgdAWQOCGlJXIrVnB0qtYwJeCV3T1KSEsl
-p6jE6+XWEE+VQpGwy2gvv5YPCpsrAQ0Drj+3fvYewRRCBH8JIeg+h8Cz191PfpW2
-gzOPzPS7EY5mX7QyIouONJOCDwLRb4Wyp342DPMNkUb5pZHcYEoN6+U343bYE7tV
-MyNMs5DfaraAPfa9nb3v0416mmGsAj0QtcJT0n7uHJClsOjbcY/TU+SAsLbMsPLr
-RsPZPEgBHTpc0PIXnJqO/Co25xxZQWat4dOC2JXurtyrcpTLC4rf73D3+eJ3dmRn
-LQLcGtsC6V5jPz4H6WXkAnhiVeFSGFqjnFbdzdP38lVFYiDO/UGTucIgkCOkQU8w
-XF6gl2+kH4dql4dw+dVWt2rM8SjSJjRXZZwtQUOLbL0Vb+zOPp9NL6Yh2PdN0WNP
-8J4Ew4IzWDQgS4ENQg1EYkgbTM/gaeTI+JyYH3Q3Oj8jQDqIgMKcm7pquSw6Z6a2
-XIP9kz9uUxlI4z369Y4DQI/mr8XeqvhwdzYrbjrfvNf2x08Cf6GtLzzpCEUvVxrx
-W5wnBzpEQz4FsUobQqSQq85L98Gb191vBRH56KOhmWwuJz2xVG/rsuj45hJvimkF
-zMb2xC2wBrYhUChQ3IyyjGXO2StRuGJWcLRGTeRMoYYxPeMHf3PjCW7upykqgCOK
-tgwinkMcqYxHLAJtE++Ip9fRYNlrKICXirf/YmsoVA2rYzqsV54vlSv0oD2hYKBX
-Ggkj7vxNm0WD8gofs3cOmCCzFvyXAzD8qz5+/D52ZMtLTclsj6pw7APFgTFDEZr8
-LbollAtzfsE5AcpocBrvpLrcE9Y3FwkNyqohlLs+Q/2MgOZY/XUjWsofCy41ws13
-DHOzL5rkA2hYyD/zdo0FU9I02wmvcO1nC/8vZDUtcSJ68vIrfW2+gL0YODcBZ6Rf
-SClSeJ+1uIBwyhBq4Sc7wGFOlvjoD0Kx+rwF7zGROKsPp+8shsGrhTZ8EJ3HCmvm
-KxHd+kXn0sM6wc1gJ2gm14dJX+N/JiAfmaj5AKYN1ROBbvwnigNYaw46xyChXCTR
-CKZaHueBPSEFLktdGBTxT6tc5YHawSz2Ev8lCZcenVFtH2MunO19hkWwNXP2n4Jl
-vH9GiOGhYKE3UCac13bz3JSJbde+phehK3EUNxdf7Vo5mfxpr7tjVcZge2TYu7dK
-n7mqiUswH5zvI3qvfjHd+90K1QQKjFdvZIDr8+5sM9JywiRook6ZiBz3O10QzE7q
-pcsAQIqpYy4YkuuxKAX+Gep7MvtjViwLIAGSowIDAQABAoIEAG18dDwatQx9As62
-wFrQ/NQwOs3S6sXWqO0knoyU639+0Duf8b7uE/Ji7nm4iG8NeEAzwXfbRAgQNuKh
-VMjgaxgKSR8dCRRQzIkgp48t46dFJvQP+A7JZ9lr5J1kIs7DAz+zawYTaW9JSVgG
-mmujIS0ayXtZ4THkSaiEZTdOVUwE7+FxVB15WvXJYAg/BFzm8HblHfEkxGppiUGx
-9ULmRAJ4TZhv7Mzeq7Ny7jEJLJpPXzgHPtE/fgshC0d5mYYpzT1I99OxsTpMtAtZ
-M8nReEwcVvjJnnDGOJZo1J2X8GXBwdVE9eXWjZmTODqbytz77S/fcaxz/amzOSIG
-dYne4SsllKS0jzfxPhz2saq26ik7vxpD7i0cRQ+I0alvgISWRTQUiHSI8UTtYJ6+
-/l7VAa/isoTZOsOmwv7M7D6U9EtjrUKwBAUHFk+0Fg2w9WELhep7/bEiQWvnOMtX
-sruWBm5zv5CxW1MuujKn2t5jHNWhQaeOIrW9V6Hew3PJsLTTHUVX2Ikc8KZLObTP
-XTtGE993cADrYYxPH8l40QCHB2t6bM2maBTRy8PlMFMsYaM7rqnftlOuXyRYLm97
-oO7F8InpD1mDND4bBKbL3UOgl1QW9dbb8KOn8pERDXD9Z3K4ewyORiyuONXxBAz2
-QiaQ4gDJWD6sEySPEHcpDjemi6mKJFtDw935JxS2brso/NHoJSVdcNxisd2irA8P
-U8jyFTrTm7cO0nKFx9vv9r5dsbkqyQnJ9V0q4/DoWvnkPElKuvTeJMw6CvTjfvc1
-xIf8m3MQJTN8pqPpIOSj8DIhQIJBEVxtcajg+vlMuefR8hTCdLzQV2QEdLGjNGph
-OhFuHrHw1b3I4nCyi9MOLzpdUDgZkaw4hPg38u71RJ6bUF6pE89Yj04/svB7SZgA
-sE8+/pZcASZl1Y0T29OM3qFJvukBWwsNij5P5SLmTXDev+DXbzUvmVLCk8nJkJ5m
-n1peGst0ApHnEzxVhU+9d5FWBdIgN7fp7V2AlFt1r0TRWUYndBn9/B6xiyfd5iyK
-LpcbRjLpQ/9b3SYQZL9t65Hc40rP35rY0luQTzVd6E/D041E0NWdpAZai4GrpMKp
-NUBXBbmwnCZEzExLi78We6aF5tG9Vh+uxyYUXlrABR98jONgYIvnWaATaw6oLgam
-B84RhWUa1f+K70wrTorUO8QNvYO2n4jpwJikn4H9pFvvzzoOB7RaA+EF4HjaVmQr
-Jm6S18c7sksYFGIJYsPrXMtgOMQVYiQ3s8LDgsWKoEQYgvgxE24AoXGlXXbALrpo
-4WvRN7qLAm3n2OQYFUK0SGoELOADQFQhbNL91i0k4wzQ56ofQCb8aTmm3q76TnbK
-4gmIUIECggIBAOm3PhMqIKb3ZEZOWYMKByb7v8seuVh6mwXmgITcZIiA2f8rTfyH
-OZanYK72FsuGZEbf1ELwNUX0JobwklA1Uq9NajaJUdfe4wDpghf282APHK1nD3m3
-FX7Rsoof/dW9+bV/IIbQDwy+1ctB20m6dejt3kWyHSXly2y2L3bk1flee8oi+VCO
-ROp4u23KJcglIYlL+cL1z76lX41v8JW3deCTL/JStmK6zuvFhY5fTlOGxsLhAY3X
-VTZ0MNUpezlBaXTixi7ilgbh7rJSpCmeno5my1hdl/5Q5U/u5F5E7ZxKxlhuVWdA
-cf9qbtolSsgJ2w8axPD3dLnQ6wsNnl4yL77TdNr8OlbwIADXHH9VzZbfEgOhP4r9
-Cwz4hIUbylJ9QGZVsEVZhvqR3M6OgqlYOmC72nLPXVYbL/Rr7cTx8t+WJ4wJRP3l
-pi6AgDZXKbMKyMIhM1IscSR2swtAFyl6g6rcAjFyG+p1IKMOQ3YOqdVVUf5UYSpo
-NBTpJ+ho3CyQJK25yT2Maex5Q91M6Rk9vmXaIs+K40t/zu8cnCIH9j3In0BrpSi7
-WUyLgzXW+7ZrW7cugC93VVgjGNbgBiDimZ5duViI+xY+3OlrsUxzSHN+YY+8H2G2
-TfzT+Mgo8lJXH9tseFBLvRKhoPbbCyCwnOJ6k2k8/iqZ2eE7vQodyW/zAoICAQDE
-yK5Mlu0k3oTVasgYRNDYKJ9uyarVLaRcN6UjzNUSta2hABzQe/LG895EucK9yVa0
-4jT7EXILYQpHnQkaYtyHtN/deYWLRObT6RMws7te2MhwzQkaNavgi449dHP38TPS
-qy3Vh6v4nMhRjkhLqNiiJoDWThtkB216t+71GxdF2l9trrcvlC6q0Z65PR4LK3B/
-3m8KmlyGN940Ovb/4kEh2EZmNwbMq+3SaXiS4R7ZvKoboyNmwvCh4t5ne+NXz3Vm
-7srej0cnOM54dTuNVdXm6pajYjELNVJGsK0vo6CAZiAOyasFkwMFMr5w/ehBaZIo
-9ewWjvgDdvQ0+JBMVZJot7e5D4tUgVTaIMpQryzAp/u3r5KC1bkcC1Zie+44CaPr
-V2FspEyYQDPJHC2tRj33lbaCWDdfdhYhv6uzlcM9ExPj+RQA0xlCuxh1pxK43H4t
-PdWvcIeIKtFRWq9gsH1/wDBaIsB5Kykq4xyvMOj3dyV+dyIO1D3GG2+ZoBt8gklk
-ieDIzCMKsXbomgm64/Ewlx9tf1fbTouSWP1+Gh0Wne3sCEtCY72ebYgjtaA0w8Xr
-HGTsEdw/rSer46uULYWXYC6VYZwlyTbPh4NSD7ATrQo6HnaKugtuya9NPfYD9S9w
-PoXPXpNKvwSq9SA/XsodHet9uU+IlKnYI41hVGaukQKCAgEAu7dAFmh5vWFqmsWH
-Q6akQ/1j+KL+v0AKj1yp0Qyreq/zZ+3CP5fk+0oIb76cZG/Wzs4sUG9aeos0/qDv
-A5kwhjipnJACul38+diTcugYZgj9QZFLbrlMfIW03xf+6tQhYlNvLpih3dzRHuYQ
-WVF4LtQO+O+sVaoSD4js6pO2AQqQZrRchwUd+S1rbS0112FrZDkvrBV+/GMbMiC8
-naOy++N8WSdx8i6Uz+3f7ZBC+vd/YTsT4ncXrBr6tdsRa9VL/GPARhwb3/9LltKl
-pAAQ3dNWc+0IjW0wIVmA9u5p4mR0lsZXtgyfA0TbMD9PpwezeJhJojk++ZgBkOkp
-UAMDMkPo0ZJQ0U8ghUxBSU0DSbB1aZDz1pCTdaGN3tOJV2Pee9NLNwhHT64kKYtx
-Cs8gZF29gQVotCY2CB0/5jVmm4qgzOsNDNiM90CiugKcLX/162Z7L/8eCOmFuP2L
-HPeYFX2MDWbXYSMeSZjFOmdgpUZYCux+9m6nljwGn7bJUMmdjCNyrQrAcPydM3v4
-pRtgd7ISz5uAID3RWWAjT3oRn2Ip9rX++44ulTpg8rdviP8FtUWWE4nHdHkMFFnu
-0i7ur7Ibki7DbQSkrp6e0watJIZXmfjBQ9wHG5CteybYKqZrofOnVubeP76t6Ffa
-wjDYcVd+WsJwaosT4sH2fc+a4V0CggIAX703N+ISHGb6f47FLTGF3jTdZJxPNgpP
-teIIwUMtwz2K7RLN8gzr0Cnj0NID0iw7uCN6HVCeiLS8uJzJSgIhQxEX/XvbIify
-mdUC1iYLmhikZGUM8ah9J/Ed0f5vUYkTaoKH+SRZ4cn9l9g+ijjqtPFAq4Myxlbq
-x9LnOm7kwsJ8vTNMwqQ4lDkKHRuEG3EV+dAxfBofu9lAjW7GrzXX9Gxxy9cnwro/
-KajQ1b0bMloc9PdJCPz87/YduIFI2XcAqoSxRNi2iHmV0ntO7vN87yzg/zqNm+Nk
-VSouCEh2lSGapTcusraUuJm4l+agcfq39p+3Lj5+ocjlBUSNh+X2JmaDx1Ctu/50
-Vu+vHtQrVMwlpneQM4Fh3ygQ7jdXbgIasyw/JQlGjqhlyYmhG3VaPrOjKQVcY2Bj
-YiYLloLdi0BV5AtR1RjfaFz4iH0xmakE5mP4K7P27KPQVOFnBAa5+EZ7/856kDkU
-fV4jnYSRQ5y/LCkv88EFEPZWNIr19Hy60mRWbbsLrW8r5/zRkRDHTpmn27EiAHvV
-Ye1T3187Gb7ae4SJgRJftc2ad60s9Z7uEvYsj8Icwwoui+1nk98V6NENuPXVb3jW
-5t8KYAlr5BbDLAWYLORZHZyYYl8r6z8FxnlUxtA/nxUSY06BUvN+J+5fnTpkXJHf
-lF2vslPK5VECggIANpTlrc5Zp9c0jPOiBQT/92LdtMR9mB76WB8fX9xn1Tm7V/cI
-MatRrFzaTvi5HqHxPWJJ5cgMYKQw8kHHk1q9U7jeexZGCICFOUjk3N2Z0aXIWrx+
-hiS1i5DzRaZNI682s436cdaS4GpZIhqjIOR4/JS15Nfu+DVqyHOevWnd0D/tGoJB
-JpiePYl2k7b8sirM3bHnRDtBiMHrHyHxzpdTLImDLHXr7C5DTeQAzzo+YMu45mNm
-utKq3SRlO2wizj0qpONgJ6cHI7pypAPalliA+hPRRJqaHmTr7LpkP3+g73hf26wK
-PoF6CLkdaN945F4Z48/3zEOP3XRIQocMA46ayA+8TsOpx9rm8L1LrF1pgmJi5Ml9
-3yV93ZZ/w4Jy0pqZ28nn3gcX6JP7qtfgioLLeyuuBphpOQXl7Ys3w1d0EEErpATU
-tRG+BS16o+FhM1iXAN8c8HrqOtyJzCpYnuoWiNc+G8BIn6ox3xmihwK4hiR6s/OM
-Mxaa6T4FF/IWIzuCezYPGgybWMXj3nRHsD7eNP170irBLwS3GdD1HWM2BpmLdJa0
-m/7QQkONhCnNFtjIalfPWMX51z/8GOKRKWH5A8PY6CxVHRg1ZfHT4TlfbdRPIG8d
-6JrWt2s4eLv9VRdM+oMifHR4qvLDgg7R8pBr2FSsVXM9U2+j+drIeAhkjHI=
+MIISKgIBAAKCBAEAxq5y/tD4hMYcrKbHffrSPietrC85Ch1Pyh9HmdzECW7Q4FKx
+FUZKaKS7WYZZRzyuKAqtZ7VCFqML5ucNzSrji1acYRQqT+VUUSDp5s3BzosqjBcg
+GH0fKcxEpf3AQI5skX8ItmEuPW4wEnU4IaysdAW/aPQqs8FTQVA9094l4xSvHhmR
+CHR7IMjhtAAv7OSXxQP8gsjFZUEmqbQb5bxYB6kYn0gmCUU3J6EwGVUZ7JgAu4v7
+QutWqyP11OB85UqodI5UqsHqDzqp8xzNY0e1SSJp88+Gz7xOviTNN7zHGB6nDrV7
+/NpI796d26DXnBvP3JwkPm7vQpI0k97i+0te5UZiiTrsVbIpNPsdn2OKG4K5iGQZ
+jbVyIShWjvWfRBrIbAXsAVKI2ATRJODrGSJJqtwEPNsUsEsz8jgiht2Z5alD8w1v
+RcrMI/eJQn4b8WIVaPgJqQlldQafBvUoxAdhJXNpuHWhukagzzqGaZI7LmAvz2n0
+nuZOuwVbbvZQQQL85a4zNq0H+opXxNcKKJ/FewRxjtBgUuLcHad8qkMAB5hO/exl
+2oU8FKFjH8jkNurgE9po4ZFOfX8wkzscIskkWR8MHXoe+Ek65ydCrVN9XNYfcV3S
+BPZCBcW9aaDF/XrQyRWu3ALf1WY3Jy2RnaLVYlTIprHMOBo3VMoRZ/U2XQvmNsac
+6uUx5sOSdj/8HH3ed9PS6I7LX+GzFvBsZhOrhd5nq32+/TPMxq5ofoBpsNtVBvBN
+XzRdewaEDWT2t61KjS6vSraDNA1QoJ+K1kQu0DeOz5MRe8OVcSInbz82/nXP/7zH
+h0I1u1kC6/GJEUZclCLS/kDxpJjicaV10QMuEphUMIF1gZaRi3RW3X+6IyJu4XXI
+Ze3fRkDFgBR1Z7Z76h2WNMbL17tWbx4sP/U0A/EAlwvMZgYELB9xBgwCF/S1FXVZ
+z9VHFIiqjGVkbYdJ5HuqoUUjlxPoRfSarEdrT+ASI19tvBZoU+TBGUvzfmRUytNP
+p/5AM34wKmaYO8MJxg3O1AzcNsfMR1bRSMWoc6wOeunAf5ybUnG+hRo5DgKMIYWj
+j6cUx5tc1JEG/J/6ZYYK9JJzhMt9JDGiWHlFI4sZk9TfXGfsfeMgiQMJHqBt7frv
+k94pbBsLqwuwrROugz145NBPqhRgzxA9k4qvyAQrHTmmf5Ty8R/0csRcEjJKG9YL
+TTrgo42VKkStJuuWzKrVX4LBVUcp4cdtZLkPEitW+Ezn0hNPZLyQzYquCezvWpVa
+37CLcU4tKCb7noRPspetwjvsDvYVAe1m/UnM/8S7eOe1b9hryJXSbsJWka6Y7k8c
+47atQzk4g/oECrB2cBdZPk8QkH1qCozifyMmywIDAQABAoIEAQC102cITqRxloEd
+fJBrecU4/sURJDJ2UDb6SkATzpdV7qWX9fV7+9UTCVcv6iUAbccQfB0QITD3WIYG
+7gOSdSLDHSOGMJc8gSr2CEInaLxqUrBAftVvs+p2LSvSIXT1X6J6USDfD/bqX1H9
+Qmt5XfaBQ9VVrsb5tP6FcnLy14Yi2IYe7NTvQPKCIXV1bwptKRLAShKoeic0sDg6
+7MrvJ7wEXrta4gN1TnPYf8puFJTiEoXxHX14U7oAa6uKA1wp13ZtHOjdURPslgp8
+pCpPJowA/06pw42jxbjmiPWKgehO6ZaHfo2W7Qn9A+XB3uQTSNylD/yXapzqdrVJ
+cpyVz/drqkt5PYSWNLNDemE0wjrdTHiRGtLL/ncRY0I8GQuFn2E2gwDmgOsdjuL8
+dh6R5PA4PpxeOT0ozdofXUIz3M///zVhVTU1URAcBmkGgjT+Zn4jj4P/v8qrG8Ej
+8+83320HgiUJnBXDTOnUXMIwEZS5r2txzY0CctEwlQmFdkQiCyL468CQ1C/OGReT
+3By+6uEm4P0v7wGIOXeqZ+JOS3zWdowsFFFud6pxUvVNXGOG4tLpOGPtszZ64ZDJ
+3Tm1mBD8ws4ql92J6jLO4v5Bc3vyHz+tV2hiBCCN+daLuYgYn3I95OsfDD5epOnq
+n+h4tFHGaMMs0uCLo2pp3XeqhR4YbbKYALWoNrgh85ld2VJBzfOEN4cj+jKIhewu
+B9Sn1Lv9QZUPZZilcjI71jMDreVHP23ZMId2sRRsYb98CCsQBRurRt/kk/IF57CB
+P3xiR0KSMAwavOD3zIulMdVXT2emypAVRhyLF/jXtVa74J3bd8Igst0CjlV6f8ok
+2j1dSg9BBo9RFAZI7P1at8jT0q+Xcly/3blAEwkRNQh6Q22PQEG1NAZAY3S38C4a
+YvqElJeehKZp7lW20W1WnnPxChzLJ75BY/nouxpoKxuCpZSV4lKjLjjm0f9i5EQy
+HS6cKXmvmwhkSFok677tmcALikK5bWdcVZacdrt6JWrVxV6EWZU/UV4dDGFN5rEF
+uRoTPjpawofw6Baz1zw9exLYo0mxxcbYtgtaho99cXXUhPkEXsLBFO3rEJJeIGkw
+P2m1XuBtYiTH1HpTwGEh/59iFnFT+Ab5u83lRw6TiMLiAsMy9lNYQxkJFvzHk7Oe
+GWmwyxFLQlrZgLLMLZAj/13YybsThWDjGMk6GmFFWG/dYw9fQbCI/aaqesZlfD+W
+J8gDWmPeu7CkUAYZ6lqIpYq3PuDRtaF6+Y0hu5UvAJJ/66q1Oh3vxBxO/8RGB7Jl
+kwrwrQlbDcaxuTdEAYcwp+cZ7vQ79r8fnQXMXyPouDTCwfroPcLHF9A4b9u/1XZi
+4kvSvnLxAoICAQDx1oImPfrmiWnAyPH3DnFjgIRe2RRmVzWBkeUy2ftwGpikGRHc
+qUCk725ZGPvZbtVSam4SHPMMbeI3m6+2cxW9J6hMg8Bwh/i3MDoH0jcCdiAcSrZ/
+be2SrgzPqpiCsu13FKDXbwuHYzp/SWc/xw4fmhrFx57Y9Xh7T9Vg2trYxM8xxdc3
+tXYsxme6xvOUTu4coINWbNDUX4G1mmWy5lq+hyI2qp0orUThD4liGZBT37zCvBn5
++vpNMeKSzLJLfk1FrWsEG+hQYmZ3SC+8FlxkqTJrRsP0n1PctlBc1OnnkqlUJna/
+D6hPFB7kXiKdEW/3NKVfIVGUHU1ErwgEX7YbfnfrSau8vmoTHecnlwBrYrqjTKF5
+FJXDDojagl2LVPOLEI2kC70Hcpv1vcmdzvmVGTH+3PD/elLCw5u/RexzU7+fWnpp
+5x3RV4NJ/kbngUNVE5fGN5CankiE4XZlUmy3CmCeRIwfuCSYcsSDb9UEASq8rI4c
+guxpNS1ZWkzVOawHEArWtykBDfudZapwOEhMSXXVk0+442B/qBLMIoBZxiAWKCQC
+WrqzKuyKSstJZ2rOno8/s60LNiF/FreBEDsztxncqoMoLeDVTsff9tWSon2xbKCr
+bicPSyhJP2+mJ046tor+LbYA3UgU5JkoLvvOdEUIy9ELwS9wMB20y1vtIwKCAgEA
+0lD22r8C3qthfxISexGSz51EHGeJOHcn3DjeDf/LO5GG4Tyb0SOorwN45to7DhL1
+SV1H3kF9IdISb7OH7x9Xf83+XOOnSCIhUTGralQ3SgtqZFuQ2zoYmUN1hcIuUSQ7
+wegS02kkhzjpsSsSul2NblRBP4SRENTEgOjuiQkpmTzgHid3uF/sg/a28mLkc/Jj
+1///oA8PUcM/FkBeD0sfL4CFDLUOWppgyKvQEGDUtngx0DEIeJhnh5Ci9qZAbs5k
+EjHT4fxyxSTZv0fXbw26P5kS+twiJzD/heKAIgX1LTUclgOQdX1seRnVVQK4+Bjx
+/TFBciiR2+aWUD9InsVC+TYPviX/5eTS3f/zt2t4YGzDZRRbRku6P+9dubwAjCcF
+WLwySkA3D4wY7zwxA9FFpUJknv2AlvuGYqBvzi4XR6iO41+wrYi9y2zH9ADEWsP5
+JGd0qK8F2vCAOkMSykPFmVc6AimbgSfY975qWUuK4GqVO2ROhdsS5OyPvWQGtSFU
+y3ewd6Z3TtjnZrtXMUqJ1favZx2aP+bhMVLyXsVP6xO/NjXtEe8CpQxkrRLY5nPl
+gp3rXEm2uACoOmUefUD6lJNDei9ujQ7aJejYH7QfcTl3Q3wALliTM1wb7mVC5Uu/
+RxPsrg063KxeUxYUuf4YyUXV7Zv+HTEPhXAmkCnN3jkCggIBAKOCGuDsSvUPtzan
+JiGZ3EQgIl9WehpvoDeIguSMyRmUgduBL7h3nsNlNsB6ydzKuOROHTmV5SZkLOzY
+EEdVLD+Fzp5c/gGXjSueKb4b/SFm6HL4eMdTdasUjojhmNfVTgSTJ+RAqhTRnD1v
+TaWrWfs34B4Ux2t9HAaThXtEvW6OMmA+mTKLN+kyA/Qei5WcgBVO2T3XuKG+KR61
+00jPwk7LYvB/Dl7wN3/LV2HCUvjCbwdi0lVXwV/r+Hq0d1/fdzNC+vX+QGYJAapo
+Xg3XOIS+eN5bLCX6i2LQySjas4LkgCPrFaI8XdmJixFwLAryPTdwmtbk2hlAbnuf
+s9XxOFkCeuXUG48zVtiyARWdXKU+TV4Xc50/s7n727+RINoRh9BgUsO1RAsDDDts
+JCLUS57pOApPK4owFECmrGj4R9P5rbd6HwX3QwvFFXkzuJDRrsqYwqc77kL7/aAK
+KFh9EA/5rjWTtwHDKMsB3afYNrfMai7CpBrW0iyXzZ96FDSeF5d22z0s/UY/Q/25
+PTE+fHNHv6m1PdONuDLXWk42jX0lHZCddys4WL7GqMMfqq5og6hLQCMbwkWwRZl9
+rHhQlxZyF8RAbQj/a+IMr8rXQrMvtbf97+u4QCRw9j3/6+b08El2HieYlFOlSRUH
+Mh4fFhUSO5Kk6yP9a6t4fL0+OVrtAoICAGhHys5VbWfY1CO8YztwUDnvAWmQEKLV
+ugbgj4GjdjKYCqe9dIjzRF1avGTCiguFbOeuf8b8872EOq31942iHbF6AwgtvLDw
+c1Kd7R9wmGV3WY5fNGju2mmL4EatLGgBLjwMBOGcPYdqubyl3chYQJsW3a9P3Nna
+lYBRfoyl5WFcpMduzFa4AT7BYp62WA4Bfm4IOcg0D2V8CNjQm5K7rb/Q2qedUnus
+FhcJ0cV9LZs354MR/OBGuPanyZXAgywt28KFwEhJ8uI7Tlggl/oxdUq5bIbEBY8n
+JVtOHjhAefurXPwo0kTfJpy5KWTYT7x8KehfttKSuReuspMhfbFzn6yPmKiwkLx9
+g+ydT0t/1xUAeLWyrTsiqzuuROumAyN66ZTA47KbOnHUBj0AvFDnP8AcNmX6zfqO
+xmg6opU5l0NpmUe2F5VDus+VpWMfyjwy9WKOhXJqLd4L55h7z2JIzj4k2NdfqAIX
+LuZQHM0fvsFdN5dWOfT21dx6Sx8d7JF88F1egDojaP7IX5w5gZcoHUXg5cGcFPJg
+WQIq3ZAOB31IskN/44dD2izaA0UpljBWANON17hPuHBtqvwhTUfPoy9uAUQhJpB8
+b7/iPt8hYYGze6YTYseYFYyevB6tSMkGkNX+VBlM6AWo1bAev7IlAkcyAEurUAlL
+XSXNoOFUSQZJAoICAQCfOKCZJwjBs72pjnz4jYYnIE7cCUQeS6Zv7z6hyKQ+QHxD
+mfwNfH15f1AP7FZ/8Vg/niHcy1Sp7LoYI4MamFLXiiTOC/1KGVICxIgibeoMcN8k
+K1M/7lazl/ca+MtYfDnFU/VyCGfZBnMPxYdUutC3IkyVhjgclCwf1jsVhS8oABhn
+f0xYPU0ek/bR2fapoGavp0ZvLZG3BYmnsTQfggDP4VmfEI8nGetsnVvJOY7hC72o
+MuCvFZeoGv1sE9Sc0zRrSeorVV7JjKlg0LpPmrJx5Cc49MpWWVU/wF4wXmaGdMic
+cYK572CTBNAdk2ORkujlqDdNV2Kug4CTaT3CM6JkyW/bdlsuaPW2B0y5tAKxpnpY
+NRHmh5wP50icG1osRFfruRJuHYCH5O72Ni/FO6yGHfgEp9YjXuKO9GaBO5HQCbN0
+qZZI+JhukELO9EOAbpKpbpFI0G3T0p9iYHd3awacEOzqxaFrEPgIQGy4OgemXqHS
+QCQcMOVI2ScW/eo+WSZkYP/qE/JT2ojbVkPwxusDZ7w+c1kN5O0wBJQaJahFkUbc
+06kqPghaHVBpVKCojtM2X8lNe3pBrD1Nhrr5NgJWq7KsttdbECeSGGBCAYtO8zPD
+HVRTFMMbRhMQ1OJiTYUCtgCWQECwrqEo14bszvPBr5HQS+EmFv/Vav86b7GIVw==
 -----END RSA PRIVATE KEY-----
--- a/mysql-test/std_data/serversan-cert.pem
+++ b/mysql-test/std_data/serversan-cert.pem
@@ -5,56 +5,110 @@ Certificate:
     Signature Algorithm: sha256WithRSAEncryption
         Issuer: CN=cacert, C=FI, ST=Helsinki, L=Helsinki, O=MariaDB
         Validity
-            Not Before: Apr 25 20:52:33 2017 GMT
-            Not After : Apr 20 20:52:33 2037 GMT
+            Not Before: Dec 16 17:57:12 2018 GMT
+            Not After : Dec 11 17:57:12 2038 GMT
         Subject: C=FI, ST=Helsinki, L=Helsinki, O=MariaDB, CN=server
         Subject Public Key Info:
             Public Key Algorithm: rsaEncryption
-                Public-Key: (1024 bit)
+                Public-Key: (4096 bit)
                 Modulus:
-                    00:a7:74:d4:2b:80:cb:96:08:2a:b9:c2:87:18:0d:
-                    69:2b:da:cf:ef:21:cb:05:d4:80:2c:f3:85:bc:78:
-                    b2:42:d9:9f:f1:dc:47:68:c5:af:5a:c9:01:f0:dd:
-                    91:cb:3a:b9:38:b2:36:6b:a3:66:ef:cd:44:0f:8f:
-                    39:57:60:ad:3b:44:33:51:c2:7f:cb:5c:8d:55:b8:
-                    1e:e8:80:e0:ed:9d:8d:10:7a:42:68:73:06:63:83:
-                    ce:db:05:5b:e1:7b:f9:0e:87:20:38:b8:11:6a:b7:
-                    59:3d:4a:ca:cb:60:e6:e1:73:d9:a2:24:4a:70:93:
-                    5e:cf:d5:04:d5:ad:ac:96:a5
+                    00:c5:9c:db:25:79:e5:65:ec:10:8f:81:bc:ef:7a:
+                    b1:a8:91:ea:eb:27:9c:2f:b9:e3:0c:69:c6:e4:c2:
+                    62:87:1d:57:1a:f0:08:5f:03:73:d4:aa:17:cf:bd:
+                    fa:9f:d6:73:90:ee:4d:74:cc:14:2f:26:b5:eb:6e:
+                    41:b2:71:10:84:9c:20:e4:c5:c9:b3:b9:68:4a:d1:
+                    3c:06:6a:c0:0c:c5:c0:b2:fa:ef:68:64:1a:94:26:
+                    31:77:70:ef:c1:8e:40:89:e4:b3:22:17:52:92:03:
+                    bb:2f:4b:3f:94:8b:24:6b:9e:7e:62:64:ef:ae:e3:
+                    38:5c:37:c3:bd:ff:ea:65:f5:fa:be:0d:9d:3c:79:
+                    5a:49:c0:f5:d4:aa:6f:56:9c:fa:85:d3:37:92:ca:
+                    9c:4b:a6:ca:a7:fb:2e:33:80:3b:05:de:12:c0:24:
+                    ea:6a:1e:57:37:9a:6f:91:bf:c0:5b:9d:8e:f6:9f:
+                    4e:4d:fe:46:31:81:e4:e4:75:f9:14:7a:db:3f:a6:
+                    fb:1d:0c:88:7c:76:60:a0:bc:92:18:ef:5e:8d:1a:
+                    4e:73:85:4c:32:a6:e4:9a:99:af:91:b6:5b:67:1e:
+                    cf:b5:fe:c1:52:74:d6:20:67:8d:6f:00:5e:af:85:
+                    ad:f6:15:03:f0:7b:d9:eb:21:2c:93:a6:fa:8e:d8:
+                    1f:86:29:40:fd:88:13:16:3e:6f:26:9d:d6:a4:a9:
+                    6a:c9:56:62:a3:94:f3:26:52:f3:6c:2c:fd:e0:97:
+                    f0:02:86:13:d6:73:db:35:3f:95:f8:26:69:ba:53:
+                    d5:8e:84:57:81:b7:be:40:c5:ce:bf:e7:67:c2:f6:
+                    99:af:27:ad:6e:c5:fe:d6:86:83:3b:51:1f:59:81:
+                    8f:42:e8:44:79:36:da:6b:be:19:65:74:dc:15:25:
+                    2c:9c:5a:50:6a:9c:e6:70:08:53:a3:2e:f7:ea:f9:
+                    56:cc:bf:64:a6:09:98:32:55:26:f4:7e:86:06:11:
+                    19:1f:eb:4e:cd:58:58:5c:fc:f6:18:80:cd:c8:84:
+                    92:55:3c:6a:10:24:db:e4:3d:e0:7f:0e:72:b8:43:
+                    f9:e4:95:47:c5:06:ed:b4:70:ee:42:7f:e4:de:d7:
+                    d4:88:ac:e6:cc:8e:69:db:fa:6d:4b:14:71:08:50:
+                    4c:3f:73:c8:55:80:30:bf:8f:3c:e4:17:22:4f:cd:
+                    9f:e1:3b:f3:16:17:ad:09:02:d9:3e:72:80:61:a7:
+                    d2:47:ef:16:c1:35:7d:a7:00:15:7b:06:d6:c9:1e:
+                    6e:7a:d1:e8:ec:0a:a3:af:e4:8e:87:7e:8e:b2:2d:
+                    a9:20:f8:5a:c1:60:53:ab:4b:ee:37:3e:ad:b6:df:
+                    ec:76:65
                 Exponent: 65537 (0x10001)
         X509v3 extensions:
             X509v3 Subject Alternative Name: 
                 DNS:localhost
     Signature Algorithm: sha256WithRSAEncryption
-         4b:78:d9:09:4c:25:cc:fb:17:8f:31:13:ac:d7:36:2d:5f:d4:
-         ce:94:84:d2:a7:fa:e2:1e:ae:b6:72:1f:01:56:0f:89:80:c0:
-         01:ba:ad:d7:cb:24:c5:25:ec:f8:35:ac:52:1b:4f:af:7c:26:
-         8d:d4:d4:91:05:21:b7:ba:3f:6b:1b:8d:1d:a5:6b:7e:7d:be:
-         2f:6a:09:83:c2:c3:6c:2f:8a:31:fa:7b:36:3f:6d:e1:62:ca:
-         a0:3c:43:b8:53:5a:4a:b3:4d:7a:cb:9c:6e:db:a4:ce:a1:95:
-         5e:26:d8:22:39:8c:34:0e:92:bd:87:a2:b1:7a:68:25:57:17:
-         b2:d8:43:3b:98:e4:80:6b:7d:3e:ab:32:82:6d:b8:80:45:83:
-         d6:55:f8:cd:31:74:17:8c:42:75:09:71:66:b9:e0:94:16:ca:
-         1d:db:1e:89:12:a1:9f:00:cb:83:99:5d:5d:28:7a:df:2a:87:
-         b5:8d:f1:9c:b9:89:2a:0d:6c:af:61:00:41:cb:03:df:99:4a:
-         fe:93:81:88:ff:47:4e:2a:b5:2b:bf:85:0f:9a:21:7b:20:58:
-         7a:1c:67:b5:8b:da:db:03:69:25:db:76:0e:f9:23:57:8d:8a:
-         47:dc:15:16:7c:2d:66:8f:6a:10:f3:b2:ea:2e:31:c6:d4:2c:
-         90:15:56:f4
+         96:d9:84:ec:ac:ee:55:83:9d:09:14:75:35:bd:fe:fc:ec:8e:
+         66:8f:e6:01:0d:01:5e:b3:78:d9:92:cb:3c:98:46:fa:3d:c3:
+         67:54:87:88:3c:24:6a:53:f4:4d:4b:97:d2:8c:e8:5c:60:f4:
+         03:34:f8:b9:10:6d:9c:42:78:4b:40:60:80:8b:dc:f6:60:20:
+         26:0b:8d:e3:e5:e7:db:6d:da:d3:f7:3a:3d:6d:82:08:20:07:
+         1c:47:3e:34:2b:37:05:4a:61:f8:c8:ec:23:dc:42:e9:39:a2:
+         12:46:76:29:74:1c:42:a2:cd:e2:27:f1:f2:af:3f:40:a9:43:
+         e1:b1:1d:04:7e:19:58:93:97:92:33:e5:e5:2e:42:1e:ed:07:
+         91:f9:51:b9:78:f0:8f:5d:95:4e:eb:f6:ff:26:92:79:24:1f:
+         fb:a9:b6:d5:03:27:10:6e:6d:53:ca:f9:f6:9b:97:00:10:aa:
+         ed:55:15:42:66:a8:22:3b:1b:92:35:12:34:06:bd:8d:c5:d5:
+         b6:da:a2:ba:ee:35:e1:b3:02:e0:38:93:b1:92:8f:40:07:97:
+         f4:52:f1:d9:f6:58:a0:cb:25:cb:3b:9a:de:26:fd:3c:bd:70:
+         e5:f4:c7:83:1c:7a:97:98:fb:bf:e8:0b:85:f0:00:78:c2:2c:
+         51:b3:e4:e5:1b:c2:8c:e6:1d:00:3e:4f:e1:29:57:9a:0a:1b:
+         98:d6:6d:a6:e3:8d:e9:c3:96:6d:58:d3:17:b1:44:7e:3d:c0:
+         0f:97:fe:6e:57:d1:51:ac:d2:7d:d8:fe:c9:d0:14:ca:79:ca:
+         49:25:01:ee:38:39:ea:f7:a1:90:bf:54:5e:e4:fc:68:a0:3e:
+         d8:f1:db:33:bb:da:30:9f:51:2d:36:43:92:b6:58:11:73:10:
+         77:23:c0:b6:13:7b:4a:d3:c2:45:f3:64:b3:d2:82:78:36:a5:
+         60:91:67:b9:a5:2f:7c:8e:02:c2:e7:60:92:73:8f:44:7b:8d:
+         68:f4:b2:75:24:5a:5f:da:1f:b1:82:2c:ea:85:25:f7:c9:1b:
+         38:0c:a7:62:9e:0b:a0:60:22:6c:dd:32:bb:9f:1c:e7:db:af:
+         0e:26:18:e6:5e:b9:4a:c3:64:a6:3a:0e:38:23:8d:1d:31:7b:
+         66:5c:1f:3f:b4:b4:fd:30:65:29:6a:7e:51:3c:c6:33:65:04:
+         88:4c:1f:aa:0b:3d:ac:ab:43:b2:a6:0e:da:88:91:ea:69:3d:
+         30:d8:64:0e:cf:b4:b2:b2:14:eb:c6:89:72:8b:6c:db:b4:17:
+         d1:8e:04:81:46:97:0d:09:ac:f4:4f:44:1a:76:f2:ab:40:d2:
+         89:fd:02:30:c4:60:bc:22
 -----BEGIN CERTIFICATE-----
-MIICuzCCAaOgAwIBAgIBBDANBgkqhkiG9w0BAQsFADBWMQ8wDQYDVQQDDAZjYWNl
+MIIFPzCCAyegAwIBAgIBBDANBgkqhkiG9w0BAQsFADBWMQ8wDQYDVQQDDAZjYWNl
 cnQxCzAJBgNVBAYTAkZJMREwDwYDVQQIDAhIZWxzaW5raTERMA8GA1UEBwwISGVs
-c2lua2kxEDAOBgNVBAoMB01hcmlhREIwHhcNMTcwNDI1MjA1MjMzWhcNMzcwNDIw
-MjA1MjMzWjBWMQswCQYDVQQGEwJGSTERMA8GA1UECAwISGVsc2lua2kxETAPBgNV
+c2lua2kxEDAOBgNVBAoMB01hcmlhREIwHhcNMTgxMjE2MTc1NzEyWhcNMzgxMjEx
+MTc1NzEyWjBWMQswCQYDVQQGEwJGSTERMA8GA1UECAwISGVsc2lua2kxETAPBgNV
 BAcMCEhlbHNpbmtpMRAwDgYDVQQKDAdNYXJpYURCMQ8wDQYDVQQDDAZzZXJ2ZXIw
-gZ8wDQYJKoZIhvcNAQEBBQADgY0AMIGJAoGBAKd01CuAy5YIKrnChxgNaSvaz+8h
-ywXUgCzzhbx4skLZn/HcR2jFr1rJAfDdkcs6uTiyNmujZu/NRA+POVdgrTtEM1HC
-f8tcjVW4HuiA4O2djRB6QmhzBmODztsFW+F7+Q6HIDi4EWq3WT1Kystg5uFz2aIk
-SnCTXs/VBNWtrJalAgMBAAGjGDAWMBQGA1UdEQQNMAuCCWxvY2FsaG9zdDANBgkq
-hkiG9w0BAQsFAAOCAQEAS3jZCUwlzPsXjzETrNc2LV/UzpSE0qf64h6utnIfAVYP
-iYDAAbqt18skxSXs+DWsUhtPr3wmjdTUkQUht7o/axuNHaVrfn2+L2oJg8LDbC+K
-Mfp7Nj9t4WLKoDxDuFNaSrNNesucbtukzqGVXibYIjmMNA6SvYeisXpoJVcXsthD
-O5jkgGt9Pqsygm24gEWD1lX4zTF0F4xCdQlxZrnglBbKHdseiRKhnwDLg5ldXSh6
-3yqHtY3xnLmJKg1sr2EAQcsD35lK/pOBiP9HTiq1K7+FD5oheyBYehxntYva2wNp
-Jdt2DvkjV42KR9wVFnwtZo9qEPOy6i4xxtQskBVW9A==
+ggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQDFnNsleeVl7BCPgbzverGo
+kerrJ5wvueMMacbkwmKHHVca8AhfA3PUqhfPvfqf1nOQ7k10zBQvJrXrbkGycRCE
+nCDkxcmzuWhK0TwGasAMxcCy+u9oZBqUJjF3cO/BjkCJ5LMiF1KSA7svSz+UiyRr
+nn5iZO+u4zhcN8O9/+pl9fq+DZ08eVpJwPXUqm9WnPqF0zeSypxLpsqn+y4zgDsF
+3hLAJOpqHlc3mm+Rv8BbnY72n05N/kYxgeTkdfkUets/pvsdDIh8dmCgvJIY716N
+Gk5zhUwypuSama+RtltnHs+1/sFSdNYgZ41vAF6vha32FQPwe9nrISyTpvqO2B+G
+KUD9iBMWPm8mndakqWrJVmKjlPMmUvNsLP3gl/AChhPWc9s1P5X4Jmm6U9WOhFeB
+t75Axc6/52fC9pmvJ61uxf7WhoM7UR9ZgY9C6ER5NtprvhlldNwVJSycWlBqnOZw
+CFOjLvfq+VbMv2SmCZgyVSb0foYGERkf607NWFhc/PYYgM3IhJJVPGoQJNvkPeB/
+DnK4Q/nklUfFBu20cO5Cf+Te19SIrObMjmnb+m1LFHEIUEw/c8hVgDC/jzzkFyJP
+zZ/hO/MWF60JAtk+coBhp9JH7xbBNX2nABV7BtbJHm560ejsCqOv5I6Hfo6yLakg
++FrBYFOrS+43Pq223+x2ZQIDAQABoxgwFjAUBgNVHREEDTALgglsb2NhbGhvc3Qw
+DQYJKoZIhvcNAQELBQADggIBAJbZhOys7lWDnQkUdTW9/vzsjmaP5gENAV6zeNmS
+yzyYRvo9w2dUh4g8JGpT9E1Ll9KM6Fxg9AM0+LkQbZxCeEtAYICL3PZgICYLjePl
+59tt2tP3Oj1tggggBxxHPjQrNwVKYfjI7CPcQuk5ohJGdil0HEKizeIn8fKvP0Cp
+Q+GxHQR+GViTl5Iz5eUuQh7tB5H5Ubl48I9dlU7r9v8mknkkH/upttUDJxBubVPK
++fablwAQqu1VFUJmqCI7G5I1EjQGvY3F1bbaorruNeGzAuA4k7GSj0AHl/RS8dn2
+WKDLJcs7mt4m/Ty9cOX0x4McepeY+7/oC4XwAHjCLFGz5OUbwozmHQA+T+EpV5oK
+G5jWbabjjenDlm1Y0xexRH49wA+X/m5X0VGs0n3Y/snQFMp5ykklAe44Oer3oZC/
+VF7k/GigPtjx2zO72jCfUS02Q5K2WBFzEHcjwLYTe0rTwkXzZLPSgng2pWCRZ7ml
+L3yOAsLnYJJzj0R7jWj0snUkWl/aH7GCLOqFJffJGzgMp2KeC6BgImzdMrufHOfb
+rw4mGOZeuUrDZKY6DjgjjR0xe2ZcHz+0tP0wZSlqflE8xjNlBIhMH6oLPayrQ7Km
+DtqIkeppPTDYZA7PtLKyFOvGiXKLbNu0F9GOBIFGlw0JrPRPRBp28qtA0on9AjDE
+YLwi
 -----END CERTIFICATE-----
--- a/mysql-test/std_data/serversan-key.pem
+++ b/mysql-test/std_data/serversan-key.pem
@@ -1,16 +1,52 @@
 -----BEGIN PRIVATE KEY-----
-MIICdgIBADANBgkqhkiG9w0BAQEFAASCAmAwggJcAgEAAoGBAKd01CuAy5YIKrnC
-hxgNaSvaz+8hywXUgCzzhbx4skLZn/HcR2jFr1rJAfDdkcs6uTiyNmujZu/NRA+P
-OVdgrTtEM1HCf8tcjVW4HuiA4O2djRB6QmhzBmODztsFW+F7+Q6HIDi4EWq3WT1K
-ystg5uFz2aIkSnCTXs/VBNWtrJalAgMBAAECgYBReSgZmmpzLroK8zhjXXMEIUv1
-3w02YvOR61HwJxEkMVn+hNxBf50XoKDPHh5nMMUZbqvHpxLYLZilsVuGxcTCPVzw
-YxTooPcJY8x61oUclI2Ls5czu/OfzoJhA9ESaFn6e4xReUFmNi8ygTMuPReZZ90T
-ZvDikonKtCCk99MSaQJBANrmlPtfY57KJ18f1TqLvqy73I1vQjffSOrK3deYbvvB
-jUJ79G9Wzj8Hje2y+XkkK+OIPcND1DnoTCTuqVazn+cCQQDD1jy8zrVg/JEPhQkS
-BM7nvm4PIb0cgTPrOhsHDIF4hbaAZnA0N4ZEJ2q7YitXfOeR98x+aH/WJOrzzhmE
-VXOTAkBQ4lK6b4zH57qUk5aeg3R5LxFX0XyOWJsA5uUB/PlFXUdtAZBYc6LR92Ci
-LDeyY4M0F+t6c12/5+3615UKzGSRAkA+SGV6utcOqGTOJcZTt7nCFFtWbqmBZkoH
-1qv/2udWWFhJj8rBoKMQC+UzAS69nVjcoI2l6kA17/nVXkfZQYAHAkEAmOHCZCVQ
-9CCYTJICvoZR2euUYdnatLN8d2/ARWjzcRDTdS82P2oscATwAsvJxsphDmbOmVWP
-Hfy1t8OOCHKYAQ==
+MIIJQwIBADANBgkqhkiG9w0BAQEFAASCCS0wggkpAgEAAoICAQDFnNsleeVl7BCP
+gbzverGokerrJ5wvueMMacbkwmKHHVca8AhfA3PUqhfPvfqf1nOQ7k10zBQvJrXr
+bkGycRCEnCDkxcmzuWhK0TwGasAMxcCy+u9oZBqUJjF3cO/BjkCJ5LMiF1KSA7sv
+Sz+UiyRrnn5iZO+u4zhcN8O9/+pl9fq+DZ08eVpJwPXUqm9WnPqF0zeSypxLpsqn
++y4zgDsF3hLAJOpqHlc3mm+Rv8BbnY72n05N/kYxgeTkdfkUets/pvsdDIh8dmCg
+vJIY716NGk5zhUwypuSama+RtltnHs+1/sFSdNYgZ41vAF6vha32FQPwe9nrISyT
+pvqO2B+GKUD9iBMWPm8mndakqWrJVmKjlPMmUvNsLP3gl/AChhPWc9s1P5X4Jmm6
+U9WOhFeBt75Axc6/52fC9pmvJ61uxf7WhoM7UR9ZgY9C6ER5NtprvhlldNwVJSyc
+WlBqnOZwCFOjLvfq+VbMv2SmCZgyVSb0foYGERkf607NWFhc/PYYgM3IhJJVPGoQ
+JNvkPeB/DnK4Q/nklUfFBu20cO5Cf+Te19SIrObMjmnb+m1LFHEIUEw/c8hVgDC/
+jzzkFyJPzZ/hO/MWF60JAtk+coBhp9JH7xbBNX2nABV7BtbJHm560ejsCqOv5I6H
+fo6yLakg+FrBYFOrS+43Pq223+x2ZQIDAQABAoICAQCwK8iLPoKuPKf+6BDCBsX/
+nCasgeGSAsxRsrNoGYuxHQ0crMUpv3AyNXmI8LtQZ+UbXfebmpyb9IzWWLMlG/s3
+3eDNUy8glLhPiv1SUzTnrhy+80sf665uSSZsbzive9w5krPIPraUCdOEtDb4fJt5
+fhD6MzYi2drTC7vr1Qzo3UQuzwBQe3dsrSP79f3QX55QVB8FNK/mGcotu5ZSv4zf
+DD33InIBLQ9gOMQiUapxgqODZj5u9bMqS7z9l9b41RZFRt4z5YqAOyCl67brl+rZ
+LIyoHDv0JClFUqHc64/M/QD8JSPAGP0UYn0Kam//VtPhwJaEfEWogyi0cXaKhk6C
+vlwEvgMG+Y5cBlFjl+aeKrbJTs645+4VAtKttN7k29Lf3KRRG3tarbumQAY4A9jn
+UvpxAqURc/FW6WjipdWyNZ9kNf7RB8J16tK4FfriscTKjs8TSS1FtQD3c5pUr40g
+8ccwllEFIQiPtXGsdmRXMsZ9Dfs4zs62KtZEj9/HLMqA4dYDXNR87ov2SnSt3aDZ
+w/nbeUU0wtYOzCVRlwV6JCDkLquDK2oobFsKuFo7dgSkpgsoBKNBpwhOd2qfKq9S
+NqBBqhhOi0jBwcHArQ1I5ewdsK0tVAOvsOKWYKxTBM2UXyvSkGM1PEh04SLSa4Sq
+6Iny5BTIXN8V3JF1Np11gQKCAQEA/1dJZ3ha+tmm6MIQBm6ZqWbJgzBYklsK7qIL
+l0pz7haT70gfv1WKkdib5FCrEnh2VPpgnYb4JO8W/diWCurbrNMZWIB+wlz6tJH5
+pg5ODdCWgIlfyxnQezVU7CTM3h9VofDHUc7ImkJbK+7Rb1bio+PximB2gtVPO1nO
+QTVpx+/JnZDqIgGFfUZv+EpUMXifm3FskFKA6i1NU/g27mOTCXwHl0cQLjDf9Ff8
+sBbl6xNvYD9dK1gecmC1+ufb14XFDUnVCL1XKWS/J4+DVcDEqYQNo4pczWZ+y988
+nwV49rqJ74R4g3zdG9zYIxSWIRSLJQ0gYpLZgAIEWxeydIQndQKCAQEAxh9tFXMp
+RmHipCKINGsp0yb3f4CtffbRJV4L/CYG6xnSD8KQcOw69Y6Ip/PCtIL0lZxB09Qx
+Z5WjpLLHW2pj3hdJTmvdS2YK3Gjt+lTWQCY3MpD8VEVsaaZSh+ZwJFXlzJLRf8pP
+U5V4V1jnS1I+XxTmdCHisTk3V3kNVHpDHc/xS4HmbQwROIy4JgAt7x/rslRq0uRL
+KE8H7pnWPqy3dw/vw8q+a+GlB8UpvJIXTtquQ625TFvSXz2dbMmozXEtec7Cf/MZ
+RcuDYEoh0o8pX/nba8aWuET9b9poj1Pyo0d7b7UOenIr1ExFmsQrZF2YG3kvQNQN
+hziuVErz1JklMQKCAQEA8Ove6q34ye9LORrCHA+HFPtS72FScOVOL1LeNOWe9OqA
+sqkOq21+zZXfoVEr2MWogQ2s03Vh9RdSTjlpeOuYUjzsEUFMWP953Z5OO4wLwaUq
+HCWCKRQ2TcpIUnd2ha5zEdUpe9rsFADAHXe8TJ1iQtj0MmhJ6bByjcFyhrxrDqcF
+SJJPF77vwQAQ7ejT/8mcbwJaOOJfmJUd/TiaO9vgBHbugqIedjbPWLEcTtmMHn27
+/MhMlHUUZT8y9YaHO5eQEbW0OAbOktPoON+EQAkehgN/CuumHHe084+ylyiGteXT
++Ba5zvaleYCqvo7SCGxlzZPRFrEQX2OH0JtPWfjHcQKCAQA6SnC7yOAx9wbJp+dP
+FOB/YY2cE+33i7DJP7xi+YU7kqM9Q89figFm0gLe91cn6QexJQIxItNOWLaG9Jq3
+CLxNUaidzBqV3nX/l1e9OUy4HwZKeGHlIW/ePbM2jNQdvLAZ2nCNGqbwPp/jgyK6
+BH6ZVvre3EFoRx9DOzhyzG4GzXF+978mAlDmxHBBrffVzRx5oWC0n7qFJ4t75gMk
++YgZHbU7c+FGuGLnwFihRJJa/vgarXhBQRrKLsSsh0CFAXGRzwVmw82BvXjv0Z/V
+5Y0U0dx+/zTFDndKe4Yb9WUEKTnmCe0ylXjTw4UPHfxuUSM5ohQHDNzWFhUkc9D9
+kflhAoIBADDfCtLHD38GjFHepvLffuOpX47NCL7yAkryoDPl+bzWVTF8c1f5xHsm
+m/hKlFqPi5E1b4PhcliSlOu1rjA+K2oCYNoTdKwYghU1djr+yUy1ajYiK1/SX31L
+dSYQN+2f3ZktgUhx3Y2BLVWKFwah2UGbNW0blPgmDa6wohiV1HeVE8RCZzz0MVwR
+xekdA78fpctFfmwi52TdJdQEBd1r+Qve38Q6Y4DzsFWu7unqphLSdHDCnvvMdARS
+LjBVyJ+go2C3wWFlhY4eMjgNLNoKp5m5XcXzKiR4ZHS0NTpJWfZU/NGtlNIIe8xo
+mdVnLYdJ/hdsJ1V5mcMwYwdhZ2ayF/k=
 -----END PRIVATE KEY-----
